<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-02-15" />

<title>Enrichment of rare variant types in outliers</title>

<script src="libR/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>
<script src="libR/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libR/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libR/datatables-binding-0.15/datatables.js"></script>
<link href="libR/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libR/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libR/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libR/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libR/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Enrichment of rare variant types in outliers</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-02-15</h4>

</div>


<pre class="r"><code># Load plotting config and functions
source(&quot;src/config.R&quot;)
source(&quot;src/functions/Integration/integrate_annotate_omics.R&quot;)
source(&quot;src/functions/variant_enrichment.R&quot;)


# Read integrated omics file 
# rp &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics_full.RDS&quot;) %&gt;% as.data.table()
rp &lt;- readRDS(snakemake@input$patient_omics) %&gt;% as.data.table()

# Subset cases with WES and RNA-seq data avaliable 
rp &lt;- rp[WES_avaliable == T &amp; RNA_seq_avaliable == T]
paste(&quot;Number of samples with WES export and RNA-seq avaliable:&quot;, uniqueN(rp$SAMPLE_ID))</code></pre>
<pre><code>## [1] &quot;Number of samples with WES export and RNA-seq avaliable: 140&quot;</code></pre>
<pre class="r"><code>#Keep only genes, with both RNA and protein measured 
rp &lt;- rp[ gene_detected == &quot;RNA and protein detected&quot;]

# Filter for the genes, detected as outliers at least once
outliers &lt;- unique(rp[outlier_class != &quot;non_outlier&quot;]$geneID)
rp &lt;- rp[geneID %in% outliers]

# Add up- down- outlier class
rp &lt;- add_up_down_class(rp, 
                        Padj_threshold =  PADJ_THRESHOLD,
                        Zscore_threshold = ZSCORE_THRESHOLD, 
                        l2FC_threshold = LOG2FC_THRESHOLD)

# Subset necessary columns
rp &lt;- rp[ , c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;, &quot;outlier_class&quot;, &quot;up_down_outlier&quot;) ]
rp &lt;- rp[!duplicated(rp), ]</code></pre>
<div id="outlier-class" class="section level1">
<h1>Outlier class</h1>
<pre class="r"><code>os1 &lt;- rp[, .N, by = outlier_class]
os2 &lt;- rp[, .N, by = up_down_outlier]
colnames(os2)[1] &lt;-&quot;outlier_class&quot; 
os &lt;- rbind(os1, os2)
rm(os1, os2)</code></pre>
<div id="number-of-outliers-per-class" class="section level3">
<h3>Number of outliers per class</h3>
<pre class="r"><code>DT::datatable(os, caption = &quot;Outlier counts&quot;,   escape = F)</code></pre>
<div id="htmlwidget-b5bfe99dceaaaca8817a" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-b5bfe99dceaaaca8817a">{"x":{"filter":"none","caption":"<caption>Outlier counts<\/caption>","data":[["1","2","3","4","5","6","7","8","9","10","11","12"],["non_outlier","RNA-only","protein-only","RNA-and-protein","non_outlier","RNA_underexpression","Protein_underexpression","RNA_Protein_underexpression","RNA_overexpression","RNA_Protein_overexpression","Protein_overexpression","RNA_over_Protein_under"],[209449,244,1380,173,209449,173,880,154,71,18,500,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>outlier_class<\/th>\n      <th>N<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>####################


# As soon as it is not allowed to share genetic data,
# only the result of the analysis could be shared.
# to reproduce the analysis please prepare the variant table (vt)
# with 3 columns: sample_Id, gene_Id and variant type: stop, coding, splice ...

# Read variant annotation 
# vt &lt;- fread(&quot;../rare_variants.tsv&quot;)
# vt &lt;- vt[ geneID %in% unique(rp$geneID), c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;, &quot;var_type&quot;)]
# rp &lt;- rp[SAMPLE_ID %in% unique(vt$SAMPLE_ID) ]
# rp &lt;- merge(vt, rp, by = c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;), all.y = T )
# rp[is.na(var_type), var_type := &quot;no rare variant&quot;]
# 
# 
# enr &lt;- reshape2::dcast(data = rp, geneID +  SAMPLE_ID +  var_type ~ up_down_outlier , fun.aggregate = cat.agg)
# 
# enrichments &lt;- data.frame()
# for (outl in unique(rp$up_down_outlier)){
#   tem_df &lt;- enr
#   setnames(tem_df, outl, &quot;outliers&quot;)
#   tem_df &lt;- tem_df[, c(&quot;geneID&quot;, &quot;SAMPLE_ID&quot;, &quot;outliers&quot;, &quot;var_type&quot; )]
#   feat &lt;- unique(tem_df$var_type)
#   tem_df &lt;- dcast(data = tem_df, geneID + SAMPLE_ID  + outliers ~ var_type, fun.aggregate = cat.agg) 
#   enrichment &lt;- enrich(tem_df, outlier= &quot;outliers&quot; ,features= feat)
#   enrichment$ci_left &lt;- enrichment$Estim - 1.96 * enrichment$Std
#   enrichment$ci_right &lt;- enrichment$Estim + 1.96 * enrichment$Std
#   enrichment$outlier_class &lt;- rep(outl, nrow(enrichment))
#   enrichments &lt;- rbind(enrichments, enrichment)
#   enr$outliers &lt;- NULL
# }
# 
# enrichments &lt;- as.data.table(enrichments)
# enrichments[, significant := 1 ]
# enrichments[enrichments$Pval &gt;= 0.05 , significant := 0.9 ]
# 
# 
# enrichments &lt;- as.data.table(enrichments)
# enrichments[outlier_class == &quot;non_outlier&quot;, priori:=Estim]
# enrichments[ outlier_class != &quot;non_outlier&quot; &amp; Pval &gt;= 0.05, priori:=Estim]
# enrichments[ outlier_class != &quot;non_outlier&quot; &amp; Pval &lt; 0.05, priori:=Estim +1]
# enrichments &lt;- enrichments[order(priori, decreasing = T), ]
# enrichments$importance &lt;- seq(1:nrow(enrichments))
# enrichments &lt;- enrichments[order(importance), ]
# 
# setnames(enrichments, c(&quot;Cat&quot;, &quot;outlier_class&quot;), c(&quot;var_type&quot;, &quot;up_down_outlier&quot;))
# write_tsv(enrichments, &quot;../enrichment_rare_vt.tsv&quot;)
# 


enrichments &lt;- fread(snakemake@input$enrichments_proportions)
# enrichments &lt;- fread(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/enrichment_proportions_variants.tsv&quot;)
enrichments &lt;- enrichments[ type == &quot;var_type&quot;]
enrichments[, var_type := factor(var_type, levels = c(&quot;no rare variant&quot;, 
                                             &quot;non_coding&quot;, &quot;synonymous&quot;, &quot;coding&quot;, &quot;frameshift&quot;, &quot;splice&quot;, &quot;stop&quot;))]

nodelist &lt;- levels(enrichments$var_type)</code></pre>
</div>
<div id="outlier-class-1" class="section level2">
<h2>Outlier class</h2>
<pre class="r"><code>ggplot(data = enrichments[up_down_outlier %in% c(&quot;RNA_underexpression&quot;, &quot;Protein_underexpression&quot;, &quot;RNA_Protein_underexpression&quot;, &quot;non_outlier&quot;)], 
       aes(x = var_type, y = Estim)) +
  geom_pointrange(aes(x = var_type, ymin = ci_left, ymax = ci_right, colour = up_down_outlier, alpha = significant), position = position_dodge(width = 1)) +
  geom_hline(yintercept = 0) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;non_coding&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;synonymous&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;coding&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;frameshift&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;splice&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;stop&quot;)-0.5) +
  theme_bw() + 
  xlab(&#39;&#39;) + 
  ylab(&#39;Log odds ratio&#39;)  + 
  scale_alpha(guide = &#39;none&#39;)+
  scale_color_manual(values = outlier_colors)+ 
  coord_flip(ylim = c(min(enrichments[enrichments$Pval &lt;0.05, &#39;ci_left&#39; ] , na.rm = T),  max(enrichments[enrichments$Pval &lt;0.05, &quot;ci_right&quot; ], na.rm = T))) </code></pre>
<p><img src="Scripts_MultiOmics_Variant_analysis_enrichment_analysis_variant_type_files/figure-html/unnamed-chunk-4-1.png" width="576" /></p>
</div>
</div>

<div id="rmd-source-code">IyctLS0KIycgdGl0bGU6IEVucmljaG1lbnQgb2YgcmFyZSB2YXJpYW50IHR5cGVzIGluIG91dGxpZXJzICAKIycgYXV0aG9yOiBzbWlybm92ZAojJyB3YjoKIycgIGlucHV0OgojJyAgLSBwYXRpZW50X29taWNzOiAnYHNtIGNvbmZpZ1siUFJPQ19EQVRBIl0gKyAiL2ludGVncmF0aW9uL3BhdGllbnRfb21pY3NfZnVsbC5SRFMiYCcKIycgIC0gZW5yaWNobWVudHNfcHJvcG9ydGlvbnM6ICdgc20gY29uZmlnWyJSQVdfREFUQSJdICsgIi9lbnJpY2htZW50X3Byb3BvcnRpb25zX3ZhcmlhbnRzLnRzdiJgJwojJyAgb3V0cHV0OgojJyBvdXRwdXQ6IAojJyAgIGh0bWxfZG9jdW1lbnQ6CiMnICAgIGNvZGVfZm9sZGluZzogaGlkZQojJyAgICBjb2RlX2Rvd25sb2FkOiBUUlVFCiMnLS0tCgoKCiMgTG9hZCBwbG90dGluZyBjb25maWcgYW5kIGZ1bmN0aW9ucwpzb3VyY2UoInNyYy9jb25maWcuUiIpCnNvdXJjZSgic3JjL2Z1bmN0aW9ucy9JbnRlZ3JhdGlvbi9pbnRlZ3JhdGVfYW5ub3RhdGVfb21pY3MuUiIpCnNvdXJjZSgic3JjL2Z1bmN0aW9ucy92YXJpYW50X2VucmljaG1lbnQuUiIpCgoKIyBSZWFkIGludGVncmF0ZWQgb21pY3MgZmlsZSAKIyBycCA8LSByZWFkUkRTKCIvcy9wcm9qZWN0L21pdG9NdWx0aU9taWNzL211bHRpT01JQ3NfaW50ZWdyYXRpb24vcHJvY2Vzc2VkX2RhdGEvaW50ZWdyYXRpb24vcGF0aWVudF9vbWljc19mdWxsLlJEUyIpICU+JSBhcy5kYXRhLnRhYmxlKCkKcnAgPC0gcmVhZFJEUyhzbmFrZW1ha2VAaW5wdXQkcGF0aWVudF9vbWljcykgJT4lIGFzLmRhdGEudGFibGUoKQoKIyBTdWJzZXQgY2FzZXMgd2l0aCBXRVMgYW5kIFJOQS1zZXEgZGF0YSBhdmFsaWFibGUgCnJwIDwtIHJwW1dFU19hdmFsaWFibGUgPT0gVCAmIFJOQV9zZXFfYXZhbGlhYmxlID09IFRdCnBhc3RlKCJOdW1iZXIgb2Ygc2FtcGxlcyB3aXRoIFdFUyBleHBvcnQgYW5kIFJOQS1zZXEgYXZhbGlhYmxlOiIsIHVuaXF1ZU4ocnAkU0FNUExFX0lEKSkKCgojS2VlcCBvbmx5IGdlbmVzLCB3aXRoIGJvdGggUk5BIGFuZCBwcm90ZWluIG1lYXN1cmVkIApycCA8LSBycFsgZ2VuZV9kZXRlY3RlZCA9PSAiUk5BIGFuZCBwcm90ZWluIGRldGVjdGVkIl0KCiMgRmlsdGVyIGZvciB0aGUgZ2VuZXMsIGRldGVjdGVkIGFzIG91dGxpZXJzIGF0IGxlYXN0IG9uY2UKb3V0bGllcnMgPC0gdW5pcXVlKHJwW291dGxpZXJfY2xhc3MgIT0gIm5vbl9vdXRsaWVyIl0kZ2VuZUlEKQpycCA8LSBycFtnZW5lSUQgJWluJSBvdXRsaWVyc10KCiMgQWRkIHVwLSBkb3duLSBvdXRsaWVyIGNsYXNzCnJwIDwtIGFkZF91cF9kb3duX2NsYXNzKHJwLCAKICAgICAgICAgICAgICAgICAgICAgICAgUGFkal90aHJlc2hvbGQgPSAgUEFESl9USFJFU0hPTEQsCiAgICAgICAgICAgICAgICAgICAgICAgIFpzY29yZV90aHJlc2hvbGQgPSBaU0NPUkVfVEhSRVNIT0xELCAKICAgICAgICAgICAgICAgICAgICAgICAgbDJGQ190aHJlc2hvbGQgPSBMT0cyRkNfVEhSRVNIT0xEKQoKIyBTdWJzZXQgbmVjZXNzYXJ5IGNvbHVtbnMKcnAgPC0gcnBbICwgYygiU0FNUExFX0lEIiwgImdlbmVJRCIsICJvdXRsaWVyX2NsYXNzIiwgInVwX2Rvd25fb3V0bGllciIpIF0KcnAgPC0gcnBbIWR1cGxpY2F0ZWQocnApLCBdCgojJyAjIE91dGxpZXIgY2xhc3MKb3MxIDwtIHJwWywgLk4sIGJ5ID0gb3V0bGllcl9jbGFzc10Kb3MyIDwtIHJwWywgLk4sIGJ5ID0gdXBfZG93bl9vdXRsaWVyXQpjb2xuYW1lcyhvczIpWzFdIDwtIm91dGxpZXJfY2xhc3MiIApvcyA8LSByYmluZChvczEsIG9zMikKcm0ob3MxLCBvczIpCiMnICMjIyBOdW1iZXIgb2Ygb3V0bGllcnMgcGVyIGNsYXNzCkRUOjpkYXRhdGFibGUob3MsIGNhcHRpb24gPSAiT3V0bGllciBjb3VudHMiLCAgIGVzY2FwZSA9IEYpCgoKCiMjIyMjIyMjIyMjIyMjIyMjIyMjCgoKIyBBcyBzb29uIGFzIGl0IGlzIG5vdCBhbGxvd2VkIHRvIHNoYXJlIGdlbmV0aWMgZGF0YSwKIyBvbmx5IHRoZSByZXN1bHQgb2YgdGhlIGFuYWx5c2lzIGNvdWxkIGJlIHNoYXJlZC4KIyB0byByZXByb2R1Y2UgdGhlIGFuYWx5c2lzIHBsZWFzZSBwcmVwYXJlIHRoZSB2YXJpYW50IHRhYmxlICh2dCkKIyB3aXRoIDMgY29sdW1uczogc2FtcGxlX0lkLCBnZW5lX0lkIGFuZCB2YXJpYW50IHR5cGU6IHN0b3AsIGNvZGluZywgc3BsaWNlIC4uLgoKIyBSZWFkIHZhcmlhbnQgYW5ub3RhdGlvbiAKIyB2dCA8LSBmcmVhZCgiLi4vcmFyZV92YXJpYW50cy50c3YiKQojIHZ0IDwtIHZ0WyBnZW5lSUQgJWluJSB1bmlxdWUocnAkZ2VuZUlEKSwgYygiU0FNUExFX0lEIiwgImdlbmVJRCIsICJ2YXJfdHlwZSIpXQojIHJwIDwtIHJwW1NBTVBMRV9JRCAlaW4lIHVuaXF1ZSh2dCRTQU1QTEVfSUQpIF0KIyBycCA8LSBtZXJnZSh2dCwgcnAsIGJ5ID0gYygiU0FNUExFX0lEIiwgImdlbmVJRCIpLCBhbGwueSA9IFQgKQojIHJwW2lzLm5hKHZhcl90eXBlKSwgdmFyX3R5cGUgOj0gIm5vIHJhcmUgdmFyaWFudCJdCiMgCiMgCiMgZW5yIDwtIHJlc2hhcGUyOjpkY2FzdChkYXRhID0gcnAsIGdlbmVJRCArICBTQU1QTEVfSUQgKyAgdmFyX3R5cGUgfiB1cF9kb3duX291dGxpZXIgLCBmdW4uYWdncmVnYXRlID0gY2F0LmFnZykKIyAKIyBlbnJpY2htZW50cyA8LSBkYXRhLmZyYW1lKCkKIyBmb3IgKG91dGwgaW4gdW5pcXVlKHJwJHVwX2Rvd25fb3V0bGllcikpewojICAgdGVtX2RmIDwtIGVucgojICAgc2V0bmFtZXModGVtX2RmLCBvdXRsLCAib3V0bGllcnMiKQojICAgdGVtX2RmIDwtIHRlbV9kZlssIGMoImdlbmVJRCIsICJTQU1QTEVfSUQiLCAib3V0bGllcnMiLCAidmFyX3R5cGUiICldCiMgICBmZWF0IDwtIHVuaXF1ZSh0ZW1fZGYkdmFyX3R5cGUpCiMgICB0ZW1fZGYgPC0gZGNhc3QoZGF0YSA9IHRlbV9kZiwgZ2VuZUlEICsgU0FNUExFX0lEICArIG91dGxpZXJzIH4gdmFyX3R5cGUsIGZ1bi5hZ2dyZWdhdGUgPSBjYXQuYWdnKSAKIyAgIGVucmljaG1lbnQgPC0gZW5yaWNoKHRlbV9kZiwgb3V0bGllcj0gIm91dGxpZXJzIiAsZmVhdHVyZXM9IGZlYXQpCiMgICBlbnJpY2htZW50JGNpX2xlZnQgPC0gZW5yaWNobWVudCRFc3RpbSAtIDEuOTYgKiBlbnJpY2htZW50JFN0ZAojICAgZW5yaWNobWVudCRjaV9yaWdodCA8LSBlbnJpY2htZW50JEVzdGltICsgMS45NiAqIGVucmljaG1lbnQkU3RkCiMgICBlbnJpY2htZW50JG91dGxpZXJfY2xhc3MgPC0gcmVwKG91dGwsIG5yb3coZW5yaWNobWVudCkpCiMgICBlbnJpY2htZW50cyA8LSByYmluZChlbnJpY2htZW50cywgZW5yaWNobWVudCkKIyAgIGVuciRvdXRsaWVycyA8LSBOVUxMCiMgfQojIAojIGVucmljaG1lbnRzIDwtIGFzLmRhdGEudGFibGUoZW5yaWNobWVudHMpCiMgZW5yaWNobWVudHNbLCBzaWduaWZpY2FudCA6PSAxIF0KIyBlbnJpY2htZW50c1tlbnJpY2htZW50cyRQdmFsID49IDAuMDUgLCBzaWduaWZpY2FudCA6PSAwLjkgXQojIAojIAojIGVucmljaG1lbnRzIDwtIGFzLmRhdGEudGFibGUoZW5yaWNobWVudHMpCiMgZW5yaWNobWVudHNbb3V0bGllcl9jbGFzcyA9PSAibm9uX291dGxpZXIiLCBwcmlvcmk6PUVzdGltXQojIGVucmljaG1lbnRzWyBvdXRsaWVyX2NsYXNzICE9ICJub25fb3V0bGllciIgJiBQdmFsID49IDAuMDUsIHByaW9yaTo9RXN0aW1dCiMgZW5yaWNobWVudHNbIG91dGxpZXJfY2xhc3MgIT0gIm5vbl9vdXRsaWVyIiAmIFB2YWwgPCAwLjA1LCBwcmlvcmk6PUVzdGltICsxXQojIGVucmljaG1lbnRzIDwtIGVucmljaG1lbnRzW29yZGVyKHByaW9yaSwgZGVjcmVhc2luZyA9IFQpLCBdCiMgZW5yaWNobWVudHMkaW1wb3J0YW5jZSA8LSBzZXEoMTpucm93KGVucmljaG1lbnRzKSkKIyBlbnJpY2htZW50cyA8LSBlbnJpY2htZW50c1tvcmRlcihpbXBvcnRhbmNlKSwgXQojIAojIHNldG5hbWVzKGVucmljaG1lbnRzLCBjKCJDYXQiLCAib3V0bGllcl9jbGFzcyIpLCBjKCJ2YXJfdHlwZSIsICJ1cF9kb3duX291dGxpZXIiKSkKIyB3cml0ZV90c3YoZW5yaWNobWVudHMsICIuLi9lbnJpY2htZW50X3JhcmVfdnQudHN2IikKIyAKCgplbnJpY2htZW50cyA8LSBmcmVhZChzbmFrZW1ha2VAaW5wdXQkZW5yaWNobWVudHNfcHJvcG9ydGlvbnMpCiMgZW5yaWNobWVudHMgPC0gZnJlYWQoIi9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9yYXdfZGF0YS9lbnJpY2htZW50X3Byb3BvcnRpb25zX3ZhcmlhbnRzLnRzdiIpCmVucmljaG1lbnRzIDwtIGVucmljaG1lbnRzWyB0eXBlID09ICJ2YXJfdHlwZSJdCmVucmljaG1lbnRzWywgdmFyX3R5cGUgOj0gZmFjdG9yKHZhcl90eXBlLCBsZXZlbHMgPSBjKCJubyByYXJlIHZhcmlhbnQiLCAKICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgICAgIm5vbl9jb2RpbmciLCAic3lub255bW91cyIsICJjb2RpbmciLCAiZnJhbWVzaGlmdCIsICJzcGxpY2UiLCAic3RvcCIpKV0KCm5vZGVsaXN0IDwtIGxldmVscyhlbnJpY2htZW50cyR2YXJfdHlwZSkKCgoKIycgIyMgT3V0bGllciBjbGFzcwojKyBmaWcud2lkdGg9NiwgZmlnLmhlaWdodD05CmdncGxvdChkYXRhID0gZW5yaWNobWVudHNbdXBfZG93bl9vdXRsaWVyICVpbiUgYygiUk5BX3VuZGVyZXhwcmVzc2lvbiIsICJQcm90ZWluX3VuZGVyZXhwcmVzc2lvbiIsICJSTkFfUHJvdGVpbl91bmRlcmV4cHJlc3Npb24iLCAibm9uX291dGxpZXIiKV0sIAogICAgICAgYWVzKHggPSB2YXJfdHlwZSwgeSA9IEVzdGltKSkgKwogIGdlb21fcG9pbnRyYW5nZShhZXMoeCA9IHZhcl90eXBlLCB5bWluID0gY2lfbGVmdCwgeW1heCA9IGNpX3JpZ2h0LCBjb2xvdXIgPSB1cF9kb3duX291dGxpZXIsIGFscGhhID0gc2lnbmlmaWNhbnQpLCBwb3NpdGlvbiA9IHBvc2l0aW9uX2RvZGdlKHdpZHRoID0gMSkpICsKICBnZW9tX2hsaW5lKHlpbnRlcmNlcHQgPSAwKSArCiAgZ2VvbV92bGluZShjb2xvciA9ICJncmF5ODAiLCBsaW5ldHlwZSA9ICJkYXNoZWQiLCB4aW50ZXJjZXB0ID0gd2hpY2gobm9kZWxpc3Q9PSJub25fY29kaW5nIiktMC41KSArCiAgZ2VvbV92bGluZShjb2xvciA9ICJncmF5ODAiLCBsaW5ldHlwZSA9ICJkYXNoZWQiLCB4aW50ZXJjZXB0ID0gd2hpY2gobm9kZWxpc3Q9PSJzeW5vbnltb3VzIiktMC41KSArCiAgZ2VvbV92bGluZShjb2xvciA9ICJncmF5ODAiLCBsaW5ldHlwZSA9ICJkYXNoZWQiLCB4aW50ZXJjZXB0ID0gd2hpY2gobm9kZWxpc3Q9PSJjb2RpbmciKS0wLjUpICsKICBnZW9tX3ZsaW5lKGNvbG9yID0gImdyYXk4MCIsIGxpbmV0eXBlID0gImRhc2hlZCIsIHhpbnRlcmNlcHQgPSB3aGljaChub2RlbGlzdD09ImZyYW1lc2hpZnQiKS0wLjUpICsKICBnZW9tX3ZsaW5lKGNvbG9yID0gImdyYXk4MCIsIGxpbmV0eXBlID0gImRhc2hlZCIsIHhpbnRlcmNlcHQgPSB3aGljaChub2RlbGlzdD09InNwbGljZSIpLTAuNSkgKwogIGdlb21fdmxpbmUoY29sb3IgPSAiZ3JheTgwIiwgbGluZXR5cGUgPSAiZGFzaGVkIiwgeGludGVyY2VwdCA9IHdoaWNoKG5vZGVsaXN0PT0ic3RvcCIpLTAuNSkgKwogIHRoZW1lX2J3KCkgKyAKICB4bGFiKCcnKSArIAogIHlsYWIoJ0xvZyBvZGRzIHJhdGlvJykgICsgCiAgc2NhbGVfYWxwaGEoZ3VpZGUgPSAnbm9uZScpKwogIHNjYWxlX2NvbG9yX21hbnVhbCh2YWx1ZXMgPSBvdXRsaWVyX2NvbG9ycykrIAogIGNvb3JkX2ZsaXAoeWxpbSA9IGMobWluKGVucmljaG1lbnRzW2VucmljaG1lbnRzJFB2YWwgPDAuMDUsICdjaV9sZWZ0JyBdICwgbmEucm0gPSBUKSwgIG1heChlbnJpY2htZW50c1tlbnJpY2htZW50cyRQdmFsIDwwLjA1LCAiY2lfcmlnaHQiIF0sIG5hLnJtID0gVCkpKSAKCgoK</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("enrichment_analysis_variant_type.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
