<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-02-25" />

<title>Figure 2c MORC2 dominant case example</title>

<script src="libR/header-attrs-2.3/header-attrs.js"></script>
<script src="libR/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Figure 2c MORC2 dominant case example</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-02-25</h4>

</div>


<pre class="r"><code>dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>source(snakemake@input$config)


# Add necessaries for gene structures
suppressPackageStartupMessages({
  stringsAsFactors=F
  library(Gviz)
  library(GenomicRanges)
  library(ggbio)
  library(EnsDb.Hsapiens.v75)
})



## Making a short cut to ensembl 
edb &lt;- EnsDb.Hsapiens.v75


# Load sample annotation
# sa &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv&#39;)
sa &lt;- fread(snakemake@input$sample_annotation)
sa &lt;- sa[USE_FOR_PROTEOMICS_PAPER == T]

# Read integrated omics file 
# rp &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics.RDS&quot;) %&gt;% as.data.table()
rp &lt;- readRDS(snakemake@input$patient_omics) %&gt;% as.data.table()


# Load patient&#39;s affected organs
patients &lt;- fread(snakemake@input$patient_organs)
# patients &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/HPO/Patients_affected_organs.tsv&#39;)


######################################################</code></pre>
<pre class="r"><code>fig_human &lt;- gganatogram(data=  patients[SAMPLE_ID == &quot;OM27390&quot;], 
                           fillOutline=&#39;white&#39;, organism=&#39;human&#39;, 
                           sex= unique(patients[SAMPLE_ID == &quot;OM27390&quot;]$sex), 
                           fill=&quot;colour&quot;)+ 
  theme_void() + 
  theme(plot.title = element_text(margin = NULL,face=&quot;bold&quot;, size = 9, hjust = 0.5),
        plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;))
fig_human</code></pre>
<p><img src="Scripts_Paper_figures_Figure2_c_MORC2_dominant_files/figure-html/unnamed-chunk-2-1.png" width="288" /></p>
<pre class="r"><code>pdf(snakemake@output$fig_human, # &quot;/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Fig2_c_human.pdf&quot;,  
    width = 3, height =5,  useDingbats=FALSE )
print(fig_human) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="r"><code>samp &lt;- rp[ SAMPLE_ID == &quot;OM27390&quot; &amp;  (outlier_class != &quot;non_outlier&quot; | causal_gene == T ) ]
fig &lt;- ggplot(samp[gene_detected == &quot;RNA and protein detected&quot; ], aes(RNA_ZSCORE, PROTEIN_ZSCORE)) +
  geom_vline(xintercept = 0, color = &quot;grey50&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey50&quot;) +
  
  geom_point(aes(color= outlier_class, shape= gene_class, size = Semantic_sim )) + 
  
  xlab(&quot;RNA zScore&quot;) + 
  ylab(&quot;Protein zScore&quot;) +
  scale_color_manual(breaks = c(&quot;non_outlier&quot;,&quot;RNA_outlier&quot;,&quot;Protein_outlier&quot;, &quot;RNA_Protein_outlier&quot;),
                     values = outlier_colors) +
  
  scale_shape_manual(values = gene_shapes) +
  
  scale_size(range = c(2.5, 6)) +
  
  geom_text_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot;  &amp; gene_class == &quot;no rare variant&quot; ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= &quot;grey20&quot;, #casual_gene
                  size= 2.5, show.legend = F)+ 
  
  geom_text_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot;  &amp; gene_class == &quot;1 rare variant&quot; &amp; ( causal_gene ==F | is.na(causal_gene)) ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= &quot;black&quot;, 
                  size= 3.1, show.legend = F) + 
  
  geom_label_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot; &amp; gene_class == &quot;rare pot. biallelic variants&quot; | causal_gene == T], 
                   mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID ), 
                   box.padding = unit(0.3, &quot;lines&quot;), colour= &quot;black&quot;, 
                   size= 4, show.legend = F ) + 
  
  coord_fixed( xlim = c( -8, 8 ),  ylim = c( -8,  8 )) +  
  theme_bw() +
  
  theme( axis.title.x= element_text( size= 10, margin = NULL,face=&quot;bold&quot;),
         axis.title.y= element_text( size= 10, margin = NULL,face=&quot;bold&quot;),
         legend.title = element_blank(),
         axis.text.x = element_text(face=&quot;bold&quot;,  size=10),
         axis.text.y = element_text(face=&quot;bold&quot;,  size=10),
         legend.position = &quot;none&quot; )  # c(0.2, 0.8)</code></pre>
<pre class="r"><code>fig</code></pre>
<p><img src="Scripts_Paper_figures_Figure2_c_MORC2_dominant_files/figure-html/unnamed-chunk-3-1.png" width="384" /></p>
<pre class="r"><code>pdf(snakemake@output$fig, # &quot;/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Fig2_c.pdf&quot;,  
    width = 4, height =4,  useDingbats=FALSE )
print(fig) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>Gene structure and variant</p>
<pre class="r"><code>autoplot(edb, ~ gene_name == &quot;MORC2&quot;, stat = &quot;reduce&quot; ,  names.expr=&quot;MORC2&quot;) + 
  theme_void() +  scale_x_reverse() + 
  geom_vline( xintercept =  31345792 , colour = &quot;darkblue&quot;)</code></pre>
<p><img src="Scripts_Paper_figures_Figure2_c_MORC2_dominant_files/figure-html/unnamed-chunk-4-1.png" width="480" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>unloadNamespace(&quot;ggbio&quot;)</code></pre>

<div id="rmd-source-code">IyctLS0KIycgdGl0bGU6IEZpZ3VyZSAyYyBNT1JDMiBkb21pbmFudCBjYXNlIGV4YW1wbGUKIycgYXV0aG9yOiBzbWlybm92ZAojJyB3YjoKIycgIGlucHV0OgojJyAgLSBjb25maWc6ICdzcmMvY29uZmlnLlInCiMnICAtIHBhdGllbnRfb21pY3M6ICdgc20gY29uZmlnWyJQUk9DX0RBVEEiXSArICIvaW50ZWdyYXRpb24vcGF0aWVudF9vbWljcy5SRFMiYCcKIycgIC0gc2FtcGxlX2Fubm90YXRpb246ICdgc20gY29uZmlnWyJBTk5PVEFUSU9OIl1gJwojJyAgLSBwYXRpZW50X29yZ2FuczogJ2BzbSBjb25maWdbIlBST0NfREFUQSJdICsgIi9IUE8vUGF0aWVudHNfYWZmZWN0ZWRfb3JnYW5zLnRzdiJgJwojJyAgb3V0cHV0OgojJyAgLSBmaWdfaHVtYW46ICdgc20gY29uZmlnWyJGSUdVUkVfRElSIl0gKyAiL0ZpZzJfY19odW1hbi5wZGYiYCcKIycgIC0gZmlnOiAnYHNtIGNvbmZpZ1siRklHVVJFX0RJUiJdICsgIi9GaWcyX2MucGRmImAnCiMnIG91dHB1dDogCiMnICAgaHRtbF9kb2N1bWVudDoKIycgICAgY29kZV9mb2xkaW5nOiBoaWRlCiMnICAgIGNvZGVfZG93bmxvYWQ6IFRSVUUKIyctLS0KCmRldi5vZmYoKQpzb3VyY2Uoc25ha2VtYWtlQGlucHV0JGNvbmZpZykKCgojIEFkZCBuZWNlc3NhcmllcyBmb3IgZ2VuZSBzdHJ1Y3R1cmVzCnN1cHByZXNzUGFja2FnZVN0YXJ0dXBNZXNzYWdlcyh7CiAgc3RyaW5nc0FzRmFjdG9ycz1GCiAgbGlicmFyeShHdml6KQogIGxpYnJhcnkoR2Vub21pY1JhbmdlcykKICBsaWJyYXJ5KGdnYmlvKQogIGxpYnJhcnkoRW5zRGIuSHNhcGllbnMudjc1KQp9KQoKCgojIyBNYWtpbmcgYSBzaG9ydCBjdXQgdG8gZW5zZW1ibCAKZWRiIDwtIEVuc0RiLkhzYXBpZW5zLnY3NQoKCiMgTG9hZCBzYW1wbGUgYW5ub3RhdGlvbgojIHNhIDwtIGZyZWFkKCcvcy9wcm9qZWN0L21pdG9NdWx0aU9taWNzL211bHRpT01JQ3NfaW50ZWdyYXRpb24vcmF3X2RhdGEvcHJvdGVvbWljc19hbm5vdGF0aW9uLnRzdicpCnNhIDwtIGZyZWFkKHNuYWtlbWFrZUBpbnB1dCRzYW1wbGVfYW5ub3RhdGlvbikKc2EgPC0gc2FbVVNFX0ZPUl9QUk9URU9NSUNTX1BBUEVSID09IFRdCgojIFJlYWQgaW50ZWdyYXRlZCBvbWljcyBmaWxlIAojIHJwIDwtIHJlYWRSRFMoIi9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9wcm9jZXNzZWRfZGF0YS9pbnRlZ3JhdGlvbi9wYXRpZW50X29taWNzLlJEUyIpICU+JSBhcy5kYXRhLnRhYmxlKCkKcnAgPC0gcmVhZFJEUyhzbmFrZW1ha2VAaW5wdXQkcGF0aWVudF9vbWljcykgJT4lIGFzLmRhdGEudGFibGUoKQoKCiMgTG9hZCBwYXRpZW50J3MgYWZmZWN0ZWQgb3JnYW5zCnBhdGllbnRzIDwtIGZyZWFkKHNuYWtlbWFrZUBpbnB1dCRwYXRpZW50X29yZ2FucykKIyBwYXRpZW50cyA8LSBmcmVhZCgnL3MvcHJvamVjdC9taXRvTXVsdGlPbWljcy9tdWx0aU9NSUNzX2ludGVncmF0aW9uL3Byb2Nlc3NlZF9kYXRhL0hQTy9QYXRpZW50c19hZmZlY3RlZF9vcmdhbnMudHN2JykKCgojIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMKIysgZmlnLndpZHRoPTMsIGZpZy5oZWlnaHQ9NQoKZmlnX2h1bWFuIDwtIGdnYW5hdG9ncmFtKGRhdGE9ICBwYXRpZW50c1tTQU1QTEVfSUQgPT0gIk9NMjczOTAiXSwgCiAgICAgICAgICAgICAgICAgICAgICAgICAgIGZpbGxPdXRsaW5lPSd3aGl0ZScsIG9yZ2FuaXNtPSdodW1hbicsIAogICAgICAgICAgICAgICAgICAgICAgICAgICBzZXg9IHVuaXF1ZShwYXRpZW50c1tTQU1QTEVfSUQgPT0gIk9NMjczOTAiXSRzZXgpLCAKICAgICAgICAgICAgICAgICAgICAgICAgICAgZmlsbD0iY29sb3VyIikrIAogIHRoZW1lX3ZvaWQoKSArIAogIHRoZW1lKHBsb3QudGl0bGUgPSBlbGVtZW50X3RleHQobWFyZ2luID0gTlVMTCxmYWNlPSJib2xkIiwgc2l6ZSA9IDksIGhqdXN0ID0gMC41KSwKICAgICAgICBwbG90Lm1hcmdpbiA9IG1hcmdpbigwLCAwLCAwLCAwLCAiY20iKSkKZmlnX2h1bWFuCgpwZGYoc25ha2VtYWtlQG91dHB1dCRmaWdfaHVtYW4sICMgIi9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9GaWd1cmVzL0ZpZzJfY19odW1hbi5wZGYiLCAgCiAgICB3aWR0aCA9IDMsIGhlaWdodCA9NSwgIHVzZURpbmdiYXRzPUZBTFNFICkKcHJpbnQoZmlnX2h1bWFuKSAKZGV2Lm9mZigpCgoKc2FtcCA8LSBycFsgU0FNUExFX0lEID09ICJPTTI3MzkwIiAmICAob3V0bGllcl9jbGFzcyAhPSAibm9uX291dGxpZXIiIHwgY2F1c2FsX2dlbmUgPT0gVCApIF0KZmlnIDwtIGdncGxvdChzYW1wW2dlbmVfZGV0ZWN0ZWQgPT0gIlJOQSBhbmQgcHJvdGVpbiBkZXRlY3RlZCIgXSwgYWVzKFJOQV9aU0NPUkUsIFBST1RFSU5fWlNDT1JFKSkgKwogIGdlb21fdmxpbmUoeGludGVyY2VwdCA9IDAsIGNvbG9yID0gImdyZXk1MCIpICsKICBnZW9tX2hsaW5lKHlpbnRlcmNlcHQgPSAwLCBjb2xvciA9ICJncmV5NTAiKSArCiAgCiAgZ2VvbV9wb2ludChhZXMoY29sb3I9IG91dGxpZXJfY2xhc3MsIHNoYXBlPSBnZW5lX2NsYXNzLCBzaXplID0gU2VtYW50aWNfc2ltICkpICsgCiAgCiAgeGxhYigiUk5BIHpTY29yZSIpICsgCiAgeWxhYigiUHJvdGVpbiB6U2NvcmUiKSArCiAgc2NhbGVfY29sb3JfbWFudWFsKGJyZWFrcyA9IGMoIm5vbl9vdXRsaWVyIiwiUk5BX291dGxpZXIiLCJQcm90ZWluX291dGxpZXIiLCAiUk5BX1Byb3RlaW5fb3V0bGllciIpLAogICAgICAgICAgICAgICAgICAgICB2YWx1ZXMgPSBvdXRsaWVyX2NvbG9ycykgKwogIAogIHNjYWxlX3NoYXBlX21hbnVhbCh2YWx1ZXMgPSBnZW5lX3NoYXBlcykgKwogIAogIHNjYWxlX3NpemUocmFuZ2UgPSBjKDIuNSwgNikpICsKICAKICBnZW9tX3RleHRfcmVwZWwoZGF0YT0gc2FtcFtnZW5lX2RldGVjdGVkID09ICJSTkEgYW5kIHByb3RlaW4gZGV0ZWN0ZWQiICAmIGdlbmVfY2xhc3MgPT0gIm5vIHJhcmUgdmFyaWFudCIgXSwgCiAgICAgICAgICAgICAgICAgIG1hcHBpbmc9YWVzKFJOQV9aU0NPUkUsIFBST1RFSU5fWlNDT1JFLCBsYWJlbD0gZ2VuZUlEKSwgY29sb3VyPSAiZ3JleTIwIiwgI2Nhc3VhbF9nZW5lCiAgICAgICAgICAgICAgICAgIHNpemU9IDIuNSwgc2hvdy5sZWdlbmQgPSBGKSsgCiAgCiAgZ2VvbV90ZXh0X3JlcGVsKGRhdGE9IHNhbXBbZ2VuZV9kZXRlY3RlZCA9PSAiUk5BIGFuZCBwcm90ZWluIGRldGVjdGVkIiAgJiBnZW5lX2NsYXNzID09ICIxIHJhcmUgdmFyaWFudCIgJiAoIGNhdXNhbF9nZW5lID09RiB8IGlzLm5hKGNhdXNhbF9nZW5lKSkgXSwgCiAgICAgICAgICAgICAgICAgIG1hcHBpbmc9YWVzKFJOQV9aU0NPUkUsIFBST1RFSU5fWlNDT1JFLCBsYWJlbD0gZ2VuZUlEKSwgY29sb3VyPSAiYmxhY2siLCAKICAgICAgICAgICAgICAgICAgc2l6ZT0gMy4xLCBzaG93LmxlZ2VuZCA9IEYpICsgCiAgCiAgZ2VvbV9sYWJlbF9yZXBlbChkYXRhPSBzYW1wW2dlbmVfZGV0ZWN0ZWQgPT0gIlJOQSBhbmQgcHJvdGVpbiBkZXRlY3RlZCIgJiBnZW5lX2NsYXNzID09ICJyYXJlIHBvdC4gYmlhbGxlbGljIHZhcmlhbnRzIiB8IGNhdXNhbF9nZW5lID09IFRdLCAKICAgICAgICAgICAgICAgICAgIG1hcHBpbmc9YWVzKFJOQV9aU0NPUkUsIFBST1RFSU5fWlNDT1JFLCBsYWJlbD0gZ2VuZUlEICksIAogICAgICAgICAgICAgICAgICAgYm94LnBhZGRpbmcgPSB1bml0KDAuMywgImxpbmVzIiksIGNvbG91cj0gImJsYWNrIiwgCiAgICAgICAgICAgICAgICAgICBzaXplPSA0LCBzaG93LmxlZ2VuZCA9IEYgKSArIAogIAogIGNvb3JkX2ZpeGVkKCB4bGltID0gYyggLTgsIDggKSwgIHlsaW0gPSBjKCAtOCwgIDggKSkgKyAgCiAgdGhlbWVfYncoKSArCiAgCiAgdGhlbWUoIGF4aXMudGl0bGUueD0gZWxlbWVudF90ZXh0KCBzaXplPSAxMCwgbWFyZ2luID0gTlVMTCxmYWNlPSJib2xkIiksCiAgICAgICAgIGF4aXMudGl0bGUueT0gZWxlbWVudF90ZXh0KCBzaXplPSAxMCwgbWFyZ2luID0gTlVMTCxmYWNlPSJib2xkIiksCiAgICAgICAgIGxlZ2VuZC50aXRsZSA9IGVsZW1lbnRfYmxhbmsoKSwKICAgICAgICAgYXhpcy50ZXh0LnggPSBlbGVtZW50X3RleHQoZmFjZT0iYm9sZCIsICBzaXplPTEwKSwKICAgICAgICAgYXhpcy50ZXh0LnkgPSBlbGVtZW50X3RleHQoZmFjZT0iYm9sZCIsICBzaXplPTEwKSwKICAgICAgICAgbGVnZW5kLnBvc2l0aW9uID0gIm5vbmUiICkgICMgYygwLjIsIDAuOCkKCiMrIGZpZy53aWR0aD00LCBmaWcuaGVpZ2h0PTQKZmlnCgpwZGYoc25ha2VtYWtlQG91dHB1dCRmaWcsICMgIi9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9GaWd1cmVzL0ZpZzJfYy5wZGYiLCAgCiAgICB3aWR0aCA9IDQsIGhlaWdodCA9NCwgIHVzZURpbmdiYXRzPUZBTFNFICkKcHJpbnQoZmlnKSAKZGV2Lm9mZigpCgoKCiMnIEdlbmUgc3RydWN0dXJlIGFuZCB2YXJpYW50IAojKyBmaWcud2lkdGg9NSwgZmlnLmhlaWdodD0yCgphdXRvcGxvdChlZGIsIH4gZ2VuZV9uYW1lID09ICJNT1JDMiIsIHN0YXQgPSAicmVkdWNlIiAsICBuYW1lcy5leHByPSJNT1JDMiIpICsgCiAgdGhlbWVfdm9pZCgpICsgIHNjYWxlX3hfcmV2ZXJzZSgpICsgCiAgZ2VvbV92bGluZSggeGludGVyY2VwdCA9ICAzMTM0NTc5MiAsIGNvbG91ciA9ICJkYXJrYmx1ZSIpCmRldi5vZmYoKQoKCnVubG9hZE5hbWVzcGFjZSgiZ2diaW8iKQoKCgo=</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Figure2_c_MORC2_dominant.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
