<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="vyepez" />

<meta name="date" content="2021-02-25" />

<title>RNAseq - proteome matching and Fig S4</title>

<script src="libR/header-attrs-2.3/header-attrs.js"></script>
<script src="libR/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>
<script src="libR/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libR/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libR/datatables-binding-0.15/datatables.js"></script>
<link href="libR/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libR/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libR/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libR/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libR/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<script src="libR/plotly-binding-4.9.2.1/plotly.js"></script>
<script src="libR/typedarray-0.1/typedarray.min.js"></script>
<link href="libR/plotly-htmlwidgets-css-1.52.2/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libR/plotly-main-1.52.2/plotly-latest.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">RNAseq - proteome matching and Fig S4</h1>
<h4 class="author">vyepez</h4>
<h4 class="date">2021-02-25</h4>

</div>


<pre class="r"><code>suppressPackageStartupMessages({
  library(data.table)
  library(magrittr)
  library(ggplot2)
  library(ggthemes)
  library(dplyr)
  library(matrixStats)
  library(OUTRIDER)
  library(mclust)
  library(LSD)
  library(plotly)
  library(pheatmap)
  library(ggplotify)
})



# Read sample annotation and subset to samples that we have either RNA or Proteome
# sa &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv&#39;)
sa &lt;- fread(snakemake@input$sample_annotation)

# Prepare annotation 
sa &lt;- sa[ !(NORMALIZATION_SAMPLE == T &amp; USE_FOR_PROTEOMICS_PAPER == F)]
sa[, rna := tstrsplit(SAMPLE_ID, &quot;.2&quot;, fixed=TRUE)]
sa[TREATMENT == &quot;HEAT&quot;, rna := tstrsplit(SAMPLE_ID, &quot;_heat&quot;, fixed=TRUE)]
sa[TREATMENT == &quot;GAL&quot;, rna := tstrsplit(SAMPLE_ID, &quot;_GAL&quot;, fixed=TRUE)]
sa[TREATMENT == &quot;DNAJC30&quot;, rna := tstrsplit(SAMPLE_ID, &quot;_T_DNAJC30&quot;, fixed=TRUE)]
sa[TREATMENT == &quot;MRPL38&quot;, rna := tstrsplit(SAMPLE_ID, &quot;_T_MRPL38&quot;, fixed=TRUE)]
sa &lt;- sa[is.na(TREATMENT) &amp; TISSUE == &quot;FIBROBLAST&quot;]
sa &lt;- sa[USE_FOR_PROTEOMICS_PAPER == T]
sa[USE_FOR_PROTEOMICS_PAPER == T, rna :=  SAMPLE_ID]

sa[ , RNA_ID := paste0(rna, &quot;R&quot;)]
sa[ , PROTEOME_ID := paste0(SAMPLE_ID, &quot;P&quot;)]
sa &lt;- sa[! (is.na(RNA_ID) &amp; is.na(PROTEOME_ID))]

# Read protein matrix, subset, log-transform and center
protein_gene_mat &lt;- fread(snakemake@input$protein_gene_mat) %&gt;% as.data.frame()
# protein_gene_mat &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_not_normalized.tsv&#39;) %&gt;% as.data.frame()
rownames(protein_gene_mat) &lt;- protein_gene_mat$geneID
protein_gene_mat$geneID &lt;- NULL
setnames(protein_gene_mat, sa$SAMPLE_ID, sa$PROTEOME_ID, skip_absent=TRUE)
protein_gene_mat &lt;- as.matrix(protein_gene_mat)
protein_gene_mat[protein_gene_mat &lt; 1e5] &lt;- 0


rna_prot_dt &lt;- sa[PROTEOME_ID %in% colnames(protein_gene_mat)  &amp; TISSUE == &#39;FIBROBLAST&#39;, .(PROTEOME_ID, RNA_ID)] 


# Subset Proteomes
protein_gene_mat &lt;- protein_gene_mat[, rna_prot_dt$PROTEOME_ID]
protein_gene_mat[protein_gene_mat &lt; 1e5] &lt;- NA
protein_gene_mat &lt;- log(protein_gene_mat + 1)
protein_gene_mat &lt;- protein_gene_mat - rowMeans2(protein_gene_mat, na.rm = T)
row.names(protein_gene_mat) &lt;- toupper(row.names(protein_gene_mat))


# Read RNA data
# ods &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/outrider/ods.Rds&quot;)
ods &lt;- readRDS(snakemake@input$ods)
counts &lt;- counts(ods, normalized = F)
counts &lt;- t(t(counts) / sizeFactors(ods))
counts[counts &lt; 50] &lt;- NA
counts &lt;- log(counts + 1) - rowMeans2(log(counts + 1), na.rm = T)

counts &lt;- as.data.frame(counts)
setnames(counts, sa$SAMPLE_ID, sa$RNA_ID, skip_absent=TRUE)
counts &lt;- as.matrix(counts)

rna_prot_dt &lt;- rna_prot_dt[RNA_ID %in% colnames(counts) ]
rna_prot_dt &lt;- rna_prot_dt[order(RNA_ID, decreasing = T)]</code></pre>
<p>Dimensions of RNA count table</p>
<pre class="r"><code>counts &lt;- counts[, unique(rna_prot_dt$RNA_ID)]
dim(counts)</code></pre>
<pre><code>## [1] 14302   146</code></pre>
<p>Dimensions of protein intensity table</p>
<pre class="r"><code>protein_gene_mat &lt;- protein_gene_mat[, rna_prot_dt$PROTEOME_ID]
dim(protein_gene_mat)</code></pre>
<pre><code>## [1] 9270  146</code></pre>
<pre class="r"><code># Read rna - protein correlation data 
# rpc &lt;- fread(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/rna_protein_cor.tsv&quot;)
rpc &lt;- fread(snakemake@input$rna_prot_cor)
# Genes with significantly correlated RNA and protein levels
rpc_sign &lt;- rpc[p.value &lt; 0.05  ] # significant == T

common_genes &lt;- intersect(row.names(counts), row.names(protein_gene_mat)) 
disp &lt;- dispersions(ods)
names(disp) &lt;- rownames(ods)
disp &lt;- disp[ common_genes]
top_disp_genes &lt;- names(head(sort(disp, decreasing = T), 1000))
common_genes &lt;- intersect(common_genes, top_disp_genes)

# Uncomment for the additional filtration for the genes with signif correlated RNA and protein levels
# common_genes &lt;- intersect(common_genes, rpc_sign$geneID)</code></pre>
<p>Number of common genes</p>
<pre class="r"><code>length(common_genes)</code></pre>
<pre><code>## [1] 1000</code></pre>
<p>Dimensions of RNA count table after the gene subset</p>
<pre class="r"><code>cm &lt;- counts[common_genes, ]
dim(cm)</code></pre>
<pre><code>## [1] 1000  146</code></pre>
<p>Dimensions of protein intensity table after the gene subset</p>
<pre class="r"><code>pm &lt;- protein_gene_mat[common_genes, ]
dim(pm)</code></pre>
<pre><code>## [1] 1000  146</code></pre>
<p>Correlation matrix</p>
<pre class="r"><code>x = sapply(1:ncol(cm), function(i){
  sapply(1:ncol(pm),
         function(j) cor.test(pm[,j], cm[,i], method = &#39;spearman&#39;)$estimate)
}
)
hist(x, main = &#39;Correlation of all RNA - Protein Permutations&#39;); abline(v = 0, col = &#39;red&#39;, lty = &#39;dashed&#39;)</code></pre>
<p><img src="Scripts_MultiOmics_Integration_RNA_proteome_match_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>colnames(x) &lt;- colnames(cm)
rownames(x) &lt;- colnames(pm)
#apply(x, 1, which.max)</code></pre>
<p>Number of times where the highest correlation was not on the annotated RNA</p>
<pre class="r"><code>sum(apply(x, 1, which.max) != 1:nrow(x))</code></pre>
<pre><code>## [1] 8</code></pre>
<div id="correlation-of-all-rna---protein-combinations" class="section level1">
<h1>Correlation of all RNA - Protein combinations</h1>
<pre class="r"><code>Fig_S4a &lt;- as.ggplot(pheatmap(x, cluster_rows=F, cluster_cols=F, 
                              main = &quot;Transcriptome - proteome correlation per sample&quot;))</code></pre>
<p><img src="Scripts_MultiOmics_Integration_RNA_proteome_match_files/figure-html/unnamed-chunk-9-1.png" width="1728" /></p>
<p>Save supplementary figure 4a</p>
<pre class="r"><code>pdf(snakemake@output$fig1, # &quot;/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Supplementary_figures/S_Fig4_a.pdf&quot;,  
    width = 15, height = 15,  useDingbats=FALSE )
print(Fig_S4a) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="r"><code># y &lt;- sapply(1:ncol(pm), function(j) cor.test(pm[,j], cm[,j], method = &#39;spearman&#39;)$estimate)
# names(y) &lt;- colnames(pm)


y &lt;- sapply(1:nrow(rna_prot_dt), function(j) cor.test(pm[,rna_prot_dt[j,]$PROTEOME_ID ], cm[,rna_prot_dt[j,]$RNA_ID], method = &#39;spearman&#39;)$estimate)
names(y) &lt;- rna_prot_dt$PROTEOME_ID </code></pre>
<div id="expectation-maximation-to-see-the-classes-separation" class="section level2">
<h2>Expectation Maximation to see the classes’ separation</h2>
<pre class="r"><code>mod &lt;- Mclust(y)
mod4 &lt;- densityMclust(y)</code></pre>
<p>Number of groups the algorithm suggests</p>
<pre class="r"><code>mod4$G</code></pre>
<pre><code>## [1] 1</code></pre>
<p>Proteome ids that belong to other cluster</p>
<pre class="r"><code>names(mod$classification[mod$classification == 1]) %&gt;% sort</code></pre>
<pre><code>## NULL</code></pre>
<p>Compute the best separation between the classes</p>
<pre class="r"><code>mus &lt;- mod4$parameters$mean
sigmas &lt;- sqrt(mod4$parameters$variance$sigmasq)
fr &lt;- function(x) {
  1/sqrt(2*pi*sigmas[1]^2) * exp(-(x-mus[1])^2/(2*sigmas[1]^2)) + 1/sqrt(2*pi*sigmas[2]^2) * exp(-(x-mus[2])^2/(2*sigmas[2]^2))
}
op &lt;- optim(.3, fr, method = &#39;Brent&#39;, lower = -.5, upper = .8)

plot(mod4, what = &quot;density&quot;, data = y, breaks = 15, xlab = &quot;Correlation of RNA - Protein annotated samples&quot;)
abline(v = op$par, col = &#39;red&#39;, lty = &#39;dashed&#39;)</code></pre>
<p><img src="Scripts_MultiOmics_Integration_RNA_proteome_match_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
</div>
<div id="find-possible-matches" class="section level2">
<h2>Find possible matches</h2>
<pre class="r"><code>dt &lt;- as.data.table(melt(x))
setnames(dt, old = c(&quot;Var1&quot;, &quot;Var2&quot;), c(&quot;Proteome_ID&quot;, &quot;RNA_ID&quot;))
dt[, max_corr := value == max(value), by = Proteome_ID]
dt[, aux := paste(Proteome_ID, RNA_ID, sep = &quot;-&quot;)]
rna_prot_dt[, aux := paste(PROTEOME_ID, RNA_ID, sep = &quot;-&quot;)]
dt[, right_annot := aux %in% rna_prot_dt$aux]
dt[, aux := NULL]
DT::datatable(dt[Proteome_ID %in% names(mod$classification[mod$classification == 1]), ][max_corr == T | right_annot == T],  options = list(pageLength = 20))</code></pre>
<div id="htmlwidget-68bd1cbeb56b65905ca0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-68bd1cbeb56b65905ca0">{"x":{"filter":"none","data":[[],[],[],[],[],[]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Proteome_ID<\/th>\n      <th>RNA_ID<\/th>\n      <th>value<\/th>\n      <th>max_corr<\/th>\n      <th>right_annot<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":20,"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,20,25,50,100]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># Create separate dataset wo confirmed matches
dtX &lt;- dt[max_corr == T &amp; right_annot == T]
dts &lt;- dt[! ( (Proteome_ID %in% dtX$Proteome_ID) | (RNA_ID %in% dtX$RNA_ID) ) ]
dts[, max_corr := value == max(value), by = Proteome_ID]
# View(dts[right_annot == T |max_corr == T])</code></pre>
</div>
<div id="plot-all-correlations-of-proteomes" class="section level2">
<h2>Plot all correlations of proteomes</h2>
<pre class="r"><code>dt[, Proteome_ID := as.character(Proteome_ID)]
plotlist = list()
for(s in unique(dt$Proteome_ID)){
  g &lt;- ggplot(dt[Proteome_ID == s], aes(reorder(RNA_ID, value), value)) + geom_point(aes(col = right_annot)) + theme_bw() + 
    theme(axis.text.x=element_blank()) + scale_color_calc() + 
    labs(title = s, x = &#39;Rank&#39;, y = &#39;RNA - Protein Correlation&#39;)
  plotlist[[s]] = ggplotly(g)
}</code></pre>
<p>Uncomment to produce the full list of figures May cause accidental browser crash</p>
<pre class="r"><code># htmltools::tagList(plotlist)</code></pre>
<p>Supplementary figure 4b</p>
<pre class="r"><code>examp &lt;- ggplot(dt[Proteome_ID == &quot;OM35261P&quot;], aes(reorder(RNA_ID, value), value)) + geom_point(aes(col = right_annot)) + theme_bw() + 
  theme(axis.text.x=element_blank()) + scale_color_calc() + 
  labs(title = &quot;OM35261P&quot;, x = &#39;Rank&#39;, y = &#39;RNA - Protein Correlation&#39;)</code></pre>
<pre class="r"><code>ggplotly(examp)</code></pre>
<div id="htmlwidget-599ba9b786e1943dde63" style="width:672px;height:480px;" class="plotly html-widget"></div>
<script type="application/json" data-for="htmlwidget-599ba9b786e1943dde63">{"x":{"data":[{"x":[74,14,24,129,136,71,20,22,88,51,122,97,117,107,113,21,47,93,143,15,54,139,57,30,46,128,32,102,89,70,39,119,25,52,69,23,145,83,79,31,19,142,37,62,61,2,95,134,41,133,49,123,10,111,27,109,98,115,121,78,105,59,66,101,29,130,16,4,124,137,67,118,140,18,106,68,99,135,127,114,1,80,45,81,58,13,11,28,94,72,33,144,82,63,141,36,7,100,44,125,91,92,126,17,120,90,112,60,3,87,96,53,108,85,138,64,26,42,104,84,8,34,40,43,77,65,110,6,103,5,12,56,48,35,132,131,38,55,116,75,73,76,9,86,50],"y":[-0.000736665946825119,-0.125335394813645,-0.0965147403512686,0.130569923022753,0.15877985609619,-0.0037425344993392,-0.107868502254925,-0.102082838832186,0.0218364255732077,-0.043173897805199,0.118169405166647,0.0507494969716367,0.0990733889929326,0.0693619024510273,0.088353744700895,-0.105465161440105,-0.0538027125862149,0.0387634543178974,0.20418018098828,-0.121260059494492,-0.0338026855650877,0.174763400558442,-0.0315043897721214,-0.0896601130507293,-0.0615861532792362,0.128846238571631,-0.0886245399899786,0.0620082146571003,0.023576311050273,-0.00493855056679185,-0.0777025287679533,0.10500806272412,-0.0950811236499527,-0.0423227996200999,-0.00857636555391719,-0.0977803809426797,0.216066532253954,0.0140957787109307,0.00778364199556678,-0.0893971957157626,-0.109025934998687,0.184914589798747,-0.0812492140745981,-0.0204043690977358,-0.0220188923420237,-0.201110508290665,0.0454870325006563,0.149358219500707,-0.0751638971324366,0.149355099860975,-0.051797901316419,0.119682378029609,-0.151917653308952,0.0847671510075602,-0.0946450139156049,0.0746784514087667,0.0548746232487475,0.0951678054321658,0.109048202055965,0.00629368804377938,0.0686035570173951,-0.026389968549903,-0.0172443352937636,0.0612436777438235,-0.0906419271445202,0.131224349298837,-0.120647393134936,-0.182901815218962,0.120715594796638,0.161220105463891,-0.0152809215289169,0.0999299794985018,0.176084306938839,-0.110137179698152,0.0687800360186195,-0.0131320421580209,0.0555820917967679,0.15507057909757,0.128390747231911,0.0929610607450628,-0.226124707537893,0.0086354274151132,-0.0689549354040216,0.0106586970393188,-0.0299123873209531,-0.131873925995399,-0.140259265100142,-0.0916486883383432,0.0454319628039862,-0.00258677364971153,-0.0867516497495561,0.205448525848574,0.0124722493471334,-0.0201400158262431,0.182655454596761,-0.0817243639879307,-0.161958363395627,0.0604153882848924,-0.0698739380336607,0.124539387992816,0.0333107630711258,0.036374588008721,0.125960429885595,-0.113300090701986,0.108934164787518,0.0303086639030729,0.0860678120377745,-0.0252657590451324,-0.190878040521474,0.0206499440507273,0.0506432851064871,-0.0376285931384496,0.0712265125267063,0.0178502703867206,0.165488166723709,-0.0197355746446151,-0.0947479954118359,-0.0743000673825465,0.0675809177439051,0.0159927343745222,-0.154851602148675,-0.0866139545360607,-0.0753923041891167,-0.0714753654269428,0.00442786270935686,-0.0181698298683363,0.0764098447773844,-0.16300184802143,0.0631107795598657,-0.172032099653701,-0.133855032411536,-0.0319636489967532,-0.052995643478775,-0.0837514076240608,0.144543240243644,0.131558985686172,-0.0779951934488488,-0.0325224467704679,0.0978568015835097,0.00106311698753741,-0.00140021531992811,0.00172763699983069,-0.154172035536265,0.0186261075681472,-0.0480972394295298],"text":["right_annot: FALSE<br />reorder(RNA_ID, value): OM99824R<br />value: -0.0007366659","right_annot: FALSE<br />reorder(RNA_ID, value): OM96588R<br />value: -0.1253353948","right_annot: FALSE<br />reorder(RNA_ID, value): OM94976R<br />value: -0.0965147404","right_annot: FALSE<br />reorder(RNA_ID, value): OM94191R<br />value:  0.1305699230","right_annot: FALSE<br />reorder(RNA_ID, value): OM93206R<br />value:  0.1587798561","right_annot: FALSE<br />reorder(RNA_ID, value): OM92985R<br />value: -0.0037425345","right_annot: FALSE<br />reorder(RNA_ID, value): OM92438R<br />value: -0.1078685023","right_annot: FALSE<br />reorder(RNA_ID, value): OM92216R<br />value: -0.1020828388","right_annot: FALSE<br />reorder(RNA_ID, value): OM91816R<br />value:  0.0218364256","right_annot: FALSE<br />reorder(RNA_ID, value): OM91786R<br />value: -0.0431738978","right_annot: FALSE<br />reorder(RNA_ID, value): OM90682R<br />value:  0.1181694052","right_annot: FALSE<br />reorder(RNA_ID, value): OM90154R<br />value:  0.0507494970","right_annot: FALSE<br />reorder(RNA_ID, value): OM88728R<br />value:  0.0990733890","right_annot: FALSE<br />reorder(RNA_ID, value): OM88470R<br />value:  0.0693619025","right_annot: FALSE<br />reorder(RNA_ID, value): OM88146R<br />value:  0.0883537447","right_annot: FALSE<br />reorder(RNA_ID, value): OM87369R<br />value: -0.1054651614","right_annot: FALSE<br />reorder(RNA_ID, value): OM85406R<br />value: -0.0538027126","right_annot: FALSE<br />reorder(RNA_ID, value): OM85279R<br />value:  0.0387634543","right_annot: FALSE<br />reorder(RNA_ID, value): OM85000R<br />value:  0.2041801810","right_annot: FALSE<br />reorder(RNA_ID, value): OM84659R<br />value: -0.1212600595","right_annot: FALSE<br />reorder(RNA_ID, value): OM84535R<br />value: -0.0338026856","right_annot: FALSE<br />reorder(RNA_ID, value): OM82277R<br />value:  0.1747634006","right_annot: FALSE<br />reorder(RNA_ID, value): OM81045R<br />value: -0.0315043898","right_annot: FALSE<br />reorder(RNA_ID, value): OM80431R<br />value: -0.0896601131","right_annot: FALSE<br />reorder(RNA_ID, value): OM79535R<br />value: -0.0615861533","right_annot: FALSE<br />reorder(RNA_ID, value): OM79510R<br />value:  0.1288462386","right_annot: FALSE<br />reorder(RNA_ID, value): OM78873R<br />value: -0.0886245400","right_annot: FALSE<br />reorder(RNA_ID, value): OM77629R<br />value:  0.0620082147","right_annot: FALSE<br />reorder(RNA_ID, value): OM77127R<br />value:  0.0235763111","right_annot: FALSE<br />reorder(RNA_ID, value): OM75740R<br />value: -0.0049385506","right_annot: FALSE<br />reorder(RNA_ID, value): OM75219_1R<br />value: -0.0777025288","right_annot: FALSE<br />reorder(RNA_ID, value): OM74919R<br />value:  0.1050080627","right_annot: FALSE<br />reorder(RNA_ID, value): OM73953R<br />value: -0.0950811236","right_annot: FALSE<br />reorder(RNA_ID, value): OM73896R<br />value: -0.0423227996","right_annot: FALSE<br />reorder(RNA_ID, value): OM73546R<br />value: -0.0085763656","right_annot: FALSE<br />reorder(RNA_ID, value): OM72697R<br />value: -0.0977803809","right_annot: FALSE<br />reorder(RNA_ID, value): OM72361R<br />value:  0.2160665323","right_annot: FALSE<br />reorder(RNA_ID, value): OM72339R<br />value:  0.0140957787","right_annot: FALSE<br />reorder(RNA_ID, value): OM71746R<br />value:  0.0077836420","right_annot: FALSE<br />reorder(RNA_ID, value): OM71616R<br />value: -0.0893971957","right_annot: FALSE<br />reorder(RNA_ID, value): OM70822R<br />value: -0.1090259350","right_annot: FALSE<br />reorder(RNA_ID, value): OM70698R<br />value:  0.1849145898","right_annot: FALSE<br />reorder(RNA_ID, value): OM69964R<br />value: -0.0812492141","right_annot: FALSE<br />reorder(RNA_ID, value): OM68633R<br />value: -0.0204043691","right_annot: FALSE<br />reorder(RNA_ID, value): OM68216R<br />value: -0.0220188923","right_annot: FALSE<br />reorder(RNA_ID, value): OM66072R<br />value: -0.2011105083","right_annot: FALSE<br />reorder(RNA_ID, value): OM65728R<br />value:  0.0454870325","right_annot: FALSE<br />reorder(RNA_ID, value): OM65708R<br />value:  0.1493582195","right_annot: FALSE<br />reorder(RNA_ID, value): OM65380R<br />value: -0.0751638971","right_annot: FALSE<br />reorder(RNA_ID, value): OM65117R<br />value:  0.1493550999","right_annot: FALSE<br />reorder(RNA_ID, value): OM63071R<br />value: -0.0517979013","right_annot: FALSE<br />reorder(RNA_ID, value): OM61986R<br />value:  0.1196823780","right_annot: FALSE<br />reorder(RNA_ID, value): OM61355R<br />value: -0.1519176533","right_annot: FALSE<br />reorder(RNA_ID, value): OM61169R<br />value:  0.0847671510","right_annot: FALSE<br />reorder(RNA_ID, value): OM60894R<br />value: -0.0946450139","right_annot: FALSE<br />reorder(RNA_ID, value): OM59244R<br />value:  0.0746784514","right_annot: FALSE<br />reorder(RNA_ID, value): OM58230R<br />value:  0.0548746232","right_annot: FALSE<br />reorder(RNA_ID, value): OM57837R<br />value:  0.0951678054","right_annot: FALSE<br />reorder(RNA_ID, value): OM57274R<br />value:  0.1090482021","right_annot: FALSE<br />reorder(RNA_ID, value): OM56706R<br />value:  0.0062936880","right_annot: FALSE<br />reorder(RNA_ID, value): OM56505R<br />value:  0.0686035570","right_annot: FALSE<br />reorder(RNA_ID, value): OM56476R<br />value: -0.0263899685","right_annot: FALSE<br />reorder(RNA_ID, value): OM55346R<br />value: -0.0172443353","right_annot: FALSE<br />reorder(RNA_ID, value): OM55264R<br />value:  0.0612436777","right_annot: FALSE<br />reorder(RNA_ID, value): OM54862R<br />value: -0.0906419271","right_annot: FALSE<br />reorder(RNA_ID, value): OM53308R<br />value:  0.1312243493","right_annot: FALSE<br />reorder(RNA_ID, value): OM52618R<br />value: -0.1206473931","right_annot: FALSE<br />reorder(RNA_ID, value): OM52277R<br />value: -0.1829018152","right_annot: FALSE<br />reorder(RNA_ID, value): OM50974R<br />value:  0.1207155948","right_annot: FALSE<br />reorder(RNA_ID, value): OM50946R<br />value:  0.1612201055","right_annot: FALSE<br />reorder(RNA_ID, value): OM49736R<br />value: -0.0152809215","right_annot: FALSE<br />reorder(RNA_ID, value): OM49333R<br />value:  0.0999299795","right_annot: FALSE<br />reorder(RNA_ID, value): OM49209R<br />value:  0.1760843069","right_annot: FALSE<br />reorder(RNA_ID, value): OM48820R<br />value: -0.1101371797","right_annot: FALSE<br />reorder(RNA_ID, value): OM48652R<br />value:  0.0687800360","right_annot: FALSE<br />reorder(RNA_ID, value): OM47664R<br />value: -0.0131320422","right_annot: FALSE<br />reorder(RNA_ID, value): OM46241R<br />value:  0.0555820918","right_annot: FALSE<br />reorder(RNA_ID, value): OM45823R<br />value:  0.1550705791","right_annot: FALSE<br />reorder(RNA_ID, value): OM45689R<br />value:  0.1283907472","right_annot: FALSE<br />reorder(RNA_ID, value): OM43933R<br />value:  0.0929610607","right_annot: FALSE<br />reorder(RNA_ID, value): OM42562R<br />value: -0.2261247075","right_annot: FALSE<br />reorder(RNA_ID, value): OM42124R<br />value:  0.0086354274","right_annot: FALSE<br />reorder(RNA_ID, value): OM41016R<br />value: -0.0689549354","right_annot: FALSE<br />reorder(RNA_ID, value): OM40565R<br />value:  0.0106586970","right_annot: FALSE<br />reorder(RNA_ID, value): OM40423R<br />value: -0.0299123873","right_annot: FALSE<br />reorder(RNA_ID, value): OM39519R<br />value: -0.1318739260","right_annot: FALSE<br />reorder(RNA_ID, value): OM38813R<br />value: -0.1402592651","right_annot: FALSE<br />reorder(RNA_ID, value): OM37464R<br />value: -0.0916486883","right_annot: FALSE<br />reorder(RNA_ID, value): OM37109R<br />value:  0.0454319628","right_annot: FALSE<br />reorder(RNA_ID, value): OM36981R<br />value: -0.0025867736","right_annot: FALSE<br />reorder(RNA_ID, value): OM36760R<br />value: -0.0867516497","right_annot: FALSE<br />reorder(RNA_ID, value): OM36694R<br />value:  0.2054485258","right_annot: FALSE<br />reorder(RNA_ID, value): OM36526R<br />value:  0.0124722493","right_annot: FALSE<br />reorder(RNA_ID, value): OM36463R<br />value: -0.0201400158","right_annot: FALSE<br />reorder(RNA_ID, value): OM36055R<br />value:  0.1826554546","right_annot: FALSE<br />reorder(RNA_ID, value): OM35654R<br />value: -0.0817243640","right_annot: FALSE<br />reorder(RNA_ID, value): OM34814R<br />value: -0.1619583634","right_annot: FALSE<br />reorder(RNA_ID, value): OM34676R<br />value:  0.0604153883","right_annot: FALSE<br />reorder(RNA_ID, value): OM33981R<br />value: -0.0698739380","right_annot: FALSE<br />reorder(RNA_ID, value): OM33917R<br />value:  0.1245393880","right_annot: FALSE<br />reorder(RNA_ID, value): OM32691R<br />value:  0.0333107631","right_annot: FALSE<br />reorder(RNA_ID, value): OM30836R<br />value:  0.0363745880","right_annot: FALSE<br />reorder(RNA_ID, value): OM30476R<br />value:  0.1259604299","right_annot: FALSE<br />reorder(RNA_ID, value): OM30085.2R<br />value: -0.1133000907","right_annot: FALSE<br />reorder(RNA_ID, value): OM29763R<br />value:  0.1089341648","right_annot: FALSE<br />reorder(RNA_ID, value): OM28784R<br />value:  0.0303086639","right_annot: FALSE<br />reorder(RNA_ID, value): OM28497R<br />value:  0.0860678120","right_annot: FALSE<br />reorder(RNA_ID, value): OM28140R<br />value: -0.0252657590","right_annot: FALSE<br />reorder(RNA_ID, value): OM27424R<br />value: -0.1908780405","right_annot: FALSE<br />reorder(RNA_ID, value): OM27390R<br />value:  0.0206499441","right_annot: FALSE<br />reorder(RNA_ID, value): OM27175R<br />value:  0.0506432851","right_annot: FALSE<br />reorder(RNA_ID, value): OM26649R<br />value: -0.0376285931","right_annot: FALSE<br />reorder(RNA_ID, value): OM25068R<br />value:  0.0712265125","right_annot: FALSE<br />reorder(RNA_ID, value): OM23417R<br />value:  0.0178502704","right_annot: FALSE<br />reorder(RNA_ID, value): OM21111R<br />value:  0.1654881667","right_annot: FALSE<br />reorder(RNA_ID, value): OM20680R<br />value: -0.0197355746","right_annot: FALSE<br />reorder(RNA_ID, value): OM20348R<br />value: -0.0947479954","right_annot: FALSE<br />reorder(RNA_ID, value): OM18763R<br />value: -0.0743000674","right_annot: FALSE<br />reorder(RNA_ID, value): OM17530R<br />value:  0.0675809177","right_annot: FALSE<br />reorder(RNA_ID, value): OM17283R<br />value:  0.0159927344","right_annot: FALSE<br />reorder(RNA_ID, value): OM16302R<br />value: -0.1548516021","right_annot: FALSE<br />reorder(RNA_ID, value): OM16162R<br />value: -0.0866139545","right_annot: FALSE<br />reorder(RNA_ID, value): OM15760R<br />value: -0.0753923042","right_annot: FALSE<br />reorder(RNA_ID, value): OM15663R<br />value: -0.0714753654","right_annot: FALSE<br />reorder(RNA_ID, value): OM15187R<br />value:  0.0044278627","right_annot: FALSE<br />reorder(RNA_ID, value): OM14461R<br />value: -0.0181698299","right_annot: FALSE<br />reorder(RNA_ID, value): OM14157R<br />value:  0.0764098448","right_annot: FALSE<br />reorder(RNA_ID, value): OM13378R<br />value: -0.1630018480","right_annot: FALSE<br />reorder(RNA_ID, value): OM12102R<br />value:  0.0631107796","right_annot: FALSE<br />reorder(RNA_ID, value): OM11815R<br />value: -0.1720320997","right_annot: FALSE<br />reorder(RNA_ID, value): OM10641R<br />value: -0.1338550324","right_annot: FALSE<br />reorder(RNA_ID, value): OM10508R<br />value: -0.0319636490","right_annot: FALSE<br />reorder(RNA_ID, value): OM07111R<br />value: -0.0529956435","right_annot: FALSE<br />reorder(RNA_ID, value): OM06865R<br />value: -0.0837514076","right_annot: FALSE<br />reorder(RNA_ID, value): OM06339R<br />value:  0.1445432402","right_annot: FALSE<br />reorder(RNA_ID, value): OM05252R<br />value:  0.1315589857","right_annot: FALSE<br />reorder(RNA_ID, value): OM05166R<br />value: -0.0779951934","right_annot: FALSE<br />reorder(RNA_ID, value): OM04338R<br />value: -0.0325224468","right_annot: FALSE<br />reorder(RNA_ID, value): OM03592R<br />value:  0.0978568016","right_annot: FALSE<br />reorder(RNA_ID, value): OM02412R<br />value:  0.0010631170","right_annot: FALSE<br />reorder(RNA_ID, value): OM02364R<br />value: -0.0014002153","right_annot: FALSE<br />reorder(RNA_ID, value): OM01282R<br />value:  0.0017276370","right_annot: FALSE<br />reorder(RNA_ID, value): OM01130R<br />value: -0.1541720355","right_annot: FALSE<br />reorder(RNA_ID, value): OM00881R<br />value:  0.0186261076","right_annot: FALSE<br />reorder(RNA_ID, value): OM00762R<br />value: -0.0480972394"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,69,134,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(0,69,134,1)"}},"hoveron":"points","name":"FALSE","legendgroup":"FALSE","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[146],"y":[0.368552036199095],"text":"right_annot:  TRUE<br />reorder(RNA_ID, value): OM35261R<br />value:  0.3685520362","type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,66,14,1)","opacity":1,"size":5.66929133858268,"symbol":"circle","line":{"width":1.88976377952756,"color":"rgba(255,66,14,1)"}},"hoveron":"points","name":"TRUE","legendgroup":"TRUE","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":43.7625570776256,"r":7.30593607305936,"b":28.4931506849315,"l":48.9497716894977},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"title":{"text":"OM35261P","font":{"color":"rgba(0,0,0,1)","family":"","size":17.5342465753425},"x":0,"xref":"paper"},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0.4,146.6],"tickmode":"array","ticktext":["OM42562R","OM66072R","OM27424R","OM52277R","OM11815R","OM13378R","OM34814R","OM16302R","OM01130R","OM61355R","OM38813R","OM10641R","OM39519R","OM96588R","OM84659R","OM52618R","OM30085.2R","OM48820R","OM70822R","OM92438R","OM87369R","OM92216R","OM72697R","OM94976R","OM73953R","OM20348R","OM60894R","OM37464R","OM54862R","OM80431R","OM71616R","OM78873R","OM36760R","OM16162R","OM06865R","OM35654R","OM69964R","OM05166R","OM75219_1R","OM15760R","OM65380R","OM18763R","OM15663R","OM33981R","OM41016R","OM79535R","OM85406R","OM07111R","OM63071R","OM00762R","OM91786R","OM73896R","OM26649R","OM84535R","OM04338R","OM10508R","OM81045R","OM40423R","OM56476R","OM28140R","OM68216R","OM68633R","OM36463R","OM20680R","OM14461R","OM55346R","OM49736R","OM47664R","OM73546R","OM75740R","OM92985R","OM36981R","OM02364R","OM99824R","OM02412R","OM01282R","OM15187R","OM56706R","OM71746R","OM42124R","OM40565R","OM36526R","OM72339R","OM17283R","OM23417R","OM00881R","OM27390R","OM91816R","OM77127R","OM28784R","OM32691R","OM30836R","OM85279R","OM37109R","OM65728R","OM27175R","OM90154R","OM58230R","OM46241R","OM34676R","OM55264R","OM77629R","OM12102R","OM17530R","OM56505R","OM48652R","OM88470R","OM25068R","OM59244R","OM14157R","OM61169R","OM28497R","OM88146R","OM43933R","OM57837R","OM03592R","OM88728R","OM49333R","OM74919R","OM29763R","OM57274R","OM90682R","OM61986R","OM50974R","OM33917R","OM30476R","OM45689R","OM79510R","OM94191R","OM53308R","OM05252R","OM06339R","OM65117R","OM65708R","OM45823R","OM93206R","OM50946R","OM21111R","OM82277R","OM49209R","OM36055R","OM70698R","OM85000R","OM36694R","OM72361R","OM35261R"],"tickvals":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,99,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146],"categoryorder":"array","categoryarray":["OM42562R","OM66072R","OM27424R","OM52277R","OM11815R","OM13378R","OM34814R","OM16302R","OM01130R","OM61355R","OM38813R","OM10641R","OM39519R","OM96588R","OM84659R","OM52618R","OM30085.2R","OM48820R","OM70822R","OM92438R","OM87369R","OM92216R","OM72697R","OM94976R","OM73953R","OM20348R","OM60894R","OM37464R","OM54862R","OM80431R","OM71616R","OM78873R","OM36760R","OM16162R","OM06865R","OM35654R","OM69964R","OM05166R","OM75219_1R","OM15760R","OM65380R","OM18763R","OM15663R","OM33981R","OM41016R","OM79535R","OM85406R","OM07111R","OM63071R","OM00762R","OM91786R","OM73896R","OM26649R","OM84535R","OM04338R","OM10508R","OM81045R","OM40423R","OM56476R","OM28140R","OM68216R","OM68633R","OM36463R","OM20680R","OM14461R","OM55346R","OM49736R","OM47664R","OM73546R","OM75740R","OM92985R","OM36981R","OM02364R","OM99824R","OM02412R","OM01282R","OM15187R","OM56706R","OM71746R","OM42124R","OM40565R","OM36526R","OM72339R","OM17283R","OM23417R","OM00881R","OM27390R","OM91816R","OM77127R","OM28784R","OM32691R","OM30836R","OM85279R","OM37109R","OM65728R","OM27175R","OM90154R","OM58230R","OM46241R","OM34676R","OM55264R","OM77629R","OM12102R","OM17530R","OM56505R","OM48652R","OM88470R","OM25068R","OM59244R","OM14157R","OM61169R","OM28497R","OM88146R","OM43933R","OM57837R","OM03592R","OM88728R","OM49333R","OM74919R","OM29763R","OM57274R","OM90682R","OM61986R","OM50974R","OM33917R","OM30476R","OM45689R","OM79510R","OM94191R","OM53308R","OM05252R","OM06339R","OM65117R","OM65708R","OM45823R","OM93206R","OM50946R","OM21111R","OM82277R","OM49209R","OM36055R","OM70698R","OM85000R","OM36694R","OM72361R","OM35261R"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":false,"tickfont":{"color":null,"family":null,"size":0},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"y","title":{"text":"Rank","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-0.255858544724742,0.398285873385944],"tickmode":"array","ticktext":["-0.2","0.0","0.2"],"tickvals":[-0.2,0,0.2],"categoryorder":"array","categoryarray":["-0.2","0.0","0.2"],"nticks":null,"ticks":"outside","tickcolor":"rgba(51,51,51,1)","ticklen":3.65296803652968,"tickwidth":0.66417600664176,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.689497716895},"tickangle":-0,"showline":false,"linecolor":null,"linewidth":0,"showgrid":true,"gridcolor":"rgba(235,235,235,1)","gridwidth":0.66417600664176,"zeroline":false,"anchor":"x","title":{"text":"RNA - Protein Correlation","font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187}},"hoverformat":".2f"},"shapes":[{"type":"rect","fillcolor":"transparent","line":{"color":"rgba(51,51,51,1)","width":0.66417600664176,"linetype":"solid"},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1}],"showlegend":true,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.88976377952756,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.689497716895},"y":0.96751968503937},"annotations":[{"text":"right_annot","x":1.02,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":14.6118721461187},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"left","yanchor":"bottom","legendTitle":true}],"hovermode":"closest","barmode":"relative"},"config":{"doubleClick":"reset","showSendToCloud":false},"source":"A","attrs":{"fdc55da8488d":{"colour":{},"x":{},"y":{},"type":"scatter"}},"cur_data":"fdc55da8488d","visdat":{"fdc55da8488d":["function (y) ","x"]},"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.2,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>pdf(snakemake@output$fig2, # &quot;/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Supplementary_figures/S_Fig4_b.pdf&quot;,  
    width = 7, height = 5,  useDingbats=FALSE )
plot(examp)
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
</div>
</div>

<div id="rmd-source-code">#'---
#' title: RNAseq - proteome matching and Fig S4
#' author: vyepez
#' wb:
#'  input: 
#'  - sample_annotation: '`sm config["ANNOTATION"]`'
#'  - protein_gene_mat: '`sm config["RAW_Protein"]`'
#'  - ods_fib: '`sm config["PROC_DATA"] + "/outrider/ods.Rds"`'
#'  - rna_prot_cor: '`sm config["PROC_DATA"] + "/integration/rna_protein_cor.tsv"`'
#'  output:
#'  - fig1: '`sm config["FIGURE_DIR"] + "/Supplementary_figures/S_Fig4_a.pdf"`'
#'  - fig2: '`sm config["FIGURE_DIR"] + "/Supplementary_figures/S_Fig4_b.pdf"`'
#' output: 
#'   html_document:
#'    code_folding: hide
#'    code_download: TRUE
#'---


suppressPackageStartupMessages({
  library(data.table)
  library(magrittr)
  library(ggplot2)
  library(ggthemes)
  library(dplyr)
  library(matrixStats)
  library(OUTRIDER)
  library(mclust)
  library(LSD)
  library(plotly)
  library(pheatmap)
  library(ggplotify)
})



# Read sample annotation and subset to samples that we have either RNA or Proteome
# sa <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv')
sa <- fread(snakemake@input$sample_annotation)

# Prepare annotation 
sa <- sa[ !(NORMALIZATION_SAMPLE == T & USE_FOR_PROTEOMICS_PAPER == F)]
sa[, rna := tstrsplit(SAMPLE_ID, ".2", fixed=TRUE)]
sa[TREATMENT == "HEAT", rna := tstrsplit(SAMPLE_ID, "_heat", fixed=TRUE)]
sa[TREATMENT == "GAL", rna := tstrsplit(SAMPLE_ID, "_GAL", fixed=TRUE)]
sa[TREATMENT == "DNAJC30", rna := tstrsplit(SAMPLE_ID, "_T_DNAJC30", fixed=TRUE)]
sa[TREATMENT == "MRPL38", rna := tstrsplit(SAMPLE_ID, "_T_MRPL38", fixed=TRUE)]
sa <- sa[is.na(TREATMENT) & TISSUE == "FIBROBLAST"]
sa <- sa[USE_FOR_PROTEOMICS_PAPER == T]
sa[USE_FOR_PROTEOMICS_PAPER == T, rna :=  SAMPLE_ID]

sa[ , RNA_ID := paste0(rna, "R")]
sa[ , PROTEOME_ID := paste0(SAMPLE_ID, "P")]
sa <- sa[! (is.na(RNA_ID) & is.na(PROTEOME_ID))]

# Read protein matrix, subset, log-transform and center
protein_gene_mat <- fread(snakemake@input$protein_gene_mat) %>% as.data.frame()
# protein_gene_mat <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_not_normalized.tsv') %>% as.data.frame()
rownames(protein_gene_mat) <- protein_gene_mat$geneID
protein_gene_mat$geneID <- NULL
setnames(protein_gene_mat, sa$SAMPLE_ID, sa$PROTEOME_ID, skip_absent=TRUE)
protein_gene_mat <- as.matrix(protein_gene_mat)
protein_gene_mat[protein_gene_mat < 1e5] <- 0


rna_prot_dt <- sa[PROTEOME_ID %in% colnames(protein_gene_mat)  & TISSUE == 'FIBROBLAST', .(PROTEOME_ID, RNA_ID)] 


# Subset Proteomes
protein_gene_mat <- protein_gene_mat[, rna_prot_dt$PROTEOME_ID]
protein_gene_mat[protein_gene_mat < 1e5] <- NA
protein_gene_mat <- log(protein_gene_mat + 1)
protein_gene_mat <- protein_gene_mat - rowMeans2(protein_gene_mat, na.rm = T)
row.names(protein_gene_mat) <- toupper(row.names(protein_gene_mat))


# Read RNA data
# ods <- readRDS("/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/outrider/ods.Rds")
ods <- readRDS(snakemake@input$ods)
counts <- counts(ods, normalized = F)
counts <- t(t(counts) / sizeFactors(ods))
counts[counts < 50] <- NA
counts <- log(counts + 1) - rowMeans2(log(counts + 1), na.rm = T)

counts <- as.data.frame(counts)
setnames(counts, sa$SAMPLE_ID, sa$RNA_ID, skip_absent=TRUE)
counts <- as.matrix(counts)

rna_prot_dt <- rna_prot_dt[RNA_ID %in% colnames(counts) ]
rna_prot_dt <- rna_prot_dt[order(RNA_ID, decreasing = T)]

#' Dimensions of RNA count table
counts <- counts[, unique(rna_prot_dt$RNA_ID)]
dim(counts)

#' Dimensions of protein intensity table
protein_gene_mat <- protein_gene_mat[, rna_prot_dt$PROTEOME_ID]
dim(protein_gene_mat)


# Read rna - protein correlation data 
# rpc <- fread("/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/rna_protein_cor.tsv")
rpc <- fread(snakemake@input$rna_prot_cor)
# Genes with significantly correlated RNA and protein levels
rpc_sign <- rpc[p.value < 0.05  ] # significant == T

common_genes <- intersect(row.names(counts), row.names(protein_gene_mat)) 
disp <- dispersions(ods)
names(disp) <- rownames(ods)
disp <- disp[ common_genes]
top_disp_genes <- names(head(sort(disp, decreasing = T), 1000))
common_genes <- intersect(common_genes, top_disp_genes)

# Uncomment for the additional filtration for the genes with signif correlated RNA and protein levels
# common_genes <- intersect(common_genes, rpc_sign$geneID)

#' Number of common genes
length(common_genes)

#' Dimensions of RNA count table after the gene subset
cm <- counts[common_genes, ]
dim(cm)

#' Dimensions of protein intensity table  after the gene subset
pm <- protein_gene_mat[common_genes, ]
dim(pm)



#' Correlation matrix 
x = sapply(1:ncol(cm), function(i){
  sapply(1:ncol(pm),
         function(j) cor.test(pm[,j], cm[,i], method = 'spearman')$estimate)
}
)
hist(x, main = 'Correlation of all RNA - Protein Permutations'); abline(v = 0, col = 'red', lty = 'dashed')


colnames(x) <- colnames(cm)
rownames(x) <- colnames(pm)
#apply(x, 1, which.max)

#' Number of times where the highest correlation was not on the annotated RNA
sum(apply(x, 1, which.max) != 1:nrow(x))

#' # Correlation of all RNA - Protein combinations
#+ fig.width=18, fig.height=18
Fig_S4a <- as.ggplot(pheatmap(x, cluster_rows=F, cluster_cols=F, 
                              main = "Transcriptome - proteome correlation per sample"))



#' Save supplementary figure 4a
pdf(snakemake@output$fig1, # "/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Supplementary_figures/S_Fig4_a.pdf",  
    width = 15, height = 15,  useDingbats=FALSE )
print(Fig_S4a) 
dev.off()


# y <- sapply(1:ncol(pm), function(j) cor.test(pm[,j], cm[,j], method = 'spearman')$estimate)
# names(y) <- colnames(pm)


y <- sapply(1:nrow(rna_prot_dt), function(j) cor.test(pm[,rna_prot_dt[j,]$PROTEOME_ID ], cm[,rna_prot_dt[j,]$RNA_ID], method = 'spearman')$estimate)
names(y) <- rna_prot_dt$PROTEOME_ID 

#' ## Expectation Maximation to see the classes' separation
mod <- Mclust(y)
mod4 <- densityMclust(y)
#' Number of groups the algorithm suggests
mod4$G

#' Proteome ids that belong to other cluster
names(mod$classification[mod$classification == 1]) %>% sort

#' Compute the best separation between the classes
mus <- mod4$parameters$mean
sigmas <- sqrt(mod4$parameters$variance$sigmasq)
fr <- function(x) {
  1/sqrt(2*pi*sigmas[1]^2) * exp(-(x-mus[1])^2/(2*sigmas[1]^2)) + 1/sqrt(2*pi*sigmas[2]^2) * exp(-(x-mus[2])^2/(2*sigmas[2]^2))
}
op <- optim(.3, fr, method = 'Brent', lower = -.5, upper = .8)

plot(mod4, what = "density", data = y, breaks = 15, xlab = "Correlation of RNA - Protein annotated samples")
abline(v = op$par, col = 'red', lty = 'dashed')


#' ## Find possible matches
dt <- as.data.table(melt(x))
setnames(dt, old = c("Var1", "Var2"), c("Proteome_ID", "RNA_ID"))
dt[, max_corr := value == max(value), by = Proteome_ID]
dt[, aux := paste(Proteome_ID, RNA_ID, sep = "-")]
rna_prot_dt[, aux := paste(PROTEOME_ID, RNA_ID, sep = "-")]
dt[, right_annot := aux %in% rna_prot_dt$aux]
dt[, aux := NULL]
DT::datatable(dt[Proteome_ID %in% names(mod$classification[mod$classification == 1]), ][max_corr == T | right_annot == T],  options = list(pageLength = 20))



# Create separate dataset wo confirmed matches
dtX <- dt[max_corr == T & right_annot == T]
dts <- dt[! ( (Proteome_ID %in% dtX$Proteome_ID) | (RNA_ID %in% dtX$RNA_ID) ) ]
dts[, max_corr := value == max(value), by = Proteome_ID]
# View(dts[right_annot == T |max_corr == T])



#' ## Plot all correlations of proteomes 
dt[, Proteome_ID := as.character(Proteome_ID)]
plotlist = list()
for(s in unique(dt$Proteome_ID)){
  g <- ggplot(dt[Proteome_ID == s], aes(reorder(RNA_ID, value), value)) + geom_point(aes(col = right_annot)) + theme_bw() + 
    theme(axis.text.x=element_blank()) + scale_color_calc() + 
    labs(title = s, x = 'Rank', y = 'RNA - Protein Correlation')
  plotlist[[s]] = ggplotly(g)
}

#' Uncomment to produce the full list of figures
#' May cause accidental browser crash 
# htmltools::tagList(plotlist)

#' Supplementary figure 4b
examp <- ggplot(dt[Proteome_ID == "OM35261P"], aes(reorder(RNA_ID, value), value)) + geom_point(aes(col = right_annot)) + theme_bw() + 
  theme(axis.text.x=element_blank()) + scale_color_calc() + 
  labs(title = "OM35261P", x = 'Rank', y = 'RNA - Protein Correlation')

#+ fig.width=7, fig.height=5
ggplotly(examp)


pdf(snakemake@output$fig2, # "/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Supplementary_figures/S_Fig4_b.pdf",  
    width = 7, height = 5,  useDingbats=FALSE )
plot(examp)
dev.off()

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("RNA_proteome_match.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
