<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-02-25" />

<title>Supplementary Fig 1f validation of positive controls and VUS</title>

<script src="libR/header-attrs-2.3/header-attrs.js"></script>
<script src="libR/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Supplementary Fig 1f validation of positive controls and VUS</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-02-25</h4>

</div>


<pre class="r"><code># load functions
source(snakemake@input$config)

dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>##########################################################
# Read integrated omics file 
# rp &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics.RDS&quot;) %&gt;% as.data.table()
rp &lt;- readRDS(snakemake@input$patient_omics) %&gt;% as.data.table()



# Load sample annotation
# sa &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/raw_data/proteomics_annotation.tsv&#39;)
sa &lt;- fread(snakemake@input$sample_annotation)
sa &lt;- sa[USE_FOR_PROTEOMICS_PAPER == T]
sa$sample_gene &lt;- paste0(sa$SAMPLE_ID, &quot;_&quot;, sa$KNOWN_MUTATION)
sa &lt;- sa[order(CATEGORY, KNOWN_MUTATION)] 


# Remove UQCRFS1 as soon as it was not detected by proteomics 
# and RNA-seq is not avaliable for the case with known pathogenic vaiant 
sa &lt;- sa[KNOWN_MUTATION != &quot;UQCRFS1&quot;]


# Subset cases with pathogenic variants (I) and VUS (II)
sa &lt;- sa[CATEGORY %in% c( &quot;I&quot;, &quot;IIa&quot;, &quot;IIb&quot;, &quot;IIc&quot;) ]
sa[CATEGORY %in% c(&quot;IIa&quot;, &quot;IIb&quot;, &quot;IIc&quot;), CATEGORY := &quot;II&quot;]
sa$CATEGORY &lt;- factor(sa$CATEGORY, c( &quot;I&quot;, &quot;II&quot;))


# Subset data for causal genes in all samples
rp &lt;- rp[ geneID %in% sa$KNOWN_MUTATION]



# ACAD9 appears in both categories, duplicate the entry
sa[KNOWN_MUTATION == &quot;ACAD9&quot; &amp; SAMPLE_ID == &quot;OM65708&quot;, KNOWN_MUTATION := &quot; ACAD9&quot;]
sa$sample_gene &lt;- paste0(sa$SAMPLE_ID, &quot;_&quot;, sa$KNOWN_MUTATION)

rp_acad9 &lt;- rp[geneID == &quot;ACAD9&quot; &amp; SAMPLE_ID != &quot;OM65728&quot;]
rp_acad9$geneID &lt;- paste0(&quot; &quot;, rp_acad9$geneID)
rp &lt;- rp[ !(geneID == &quot;ACAD9&quot; &amp; SAMPLE_ID == &quot;OM65708&quot;) ]
rp &lt;- rbind(rp, rp_acad9 )
rp$sample_gene &lt;- paste0(rp$SAMPLE_ID, &quot;_&quot;, rp$geneID)
rm(rp_acad9)


# Annotate with category
gene_cat &lt;- sa[ , c(&quot;KNOWN_MUTATION&quot;, &quot;CATEGORY&quot;)]
gene_cat &lt;- gene_cat[!duplicated(gene_cat)]
rp &lt;- merge(rp, gene_cat, by.x = &quot;geneID&quot;, by.y = &quot;KNOWN_MUTATION&quot;)
rm(gene_cat)


# Rename categories 
rp[CATEGORY == &quot;I&quot;, CATEGORY := &quot;Pathogenic variants&quot;]
rp[CATEGORY == &quot;II&quot;, CATEGORY := &quot;VUS&quot;]


# Annotate with causal gene
rp$causal_gene &lt;- NULL
rp[, causal_gene :=  sample_gene %in% sa$sample_gene]
rp[, cat3 := sample_gene %in% sa[CATEGORY == &quot;III&quot;]$sample_gene ]

rp[geneID == &quot;STIM1&quot;, geneID := &quot;STIM1*&quot;]
rp[causal_gene == T &amp; is.na(PROTEIN_ZSCORE), PROTEIN_ZSCORE := -Inf]
rp[causal_gene == T &amp; is.na(RNA_ZSCORE), RNA_ZSCORE := -Inf]

rp_c &lt;- rp[causal_gene == T &amp; CATEGORY %in% c(&quot;Pathogenic variants&quot; ,  &quot;VUS&quot;)]
rp$geneID &lt;- factor(rp$geneID, unique(rp_c[ order(PROTEIN_ZSCORE)]$geneID))


#####################################################################</code></pre>
<pre class="r"><code>protrider &lt;- ggplot(rp, aes(geneID,  PROTEIN_ZSCORE ))+
  geom_hline(yintercept = 0, color = &quot;grey50&quot;) +
  geom_quasirandom(data = rp[causal_gene != T ] , aes(geneID,  PROTEIN_ZSCORE),color = &quot;grey80&quot; , size=0.5)+
  stat_summary(fun.data = mean_sdl, fun.args = list(mult=1), geom=&quot;errorbar&quot;, color=&quot;black&quot;, width=0.2) +
  stat_summary(fun = mean, geom=&quot;point&quot;, color=&quot;black&quot;) +
  ylab(&quot;PROTRIDER \n Protein zScore&quot;) + 
  geom_point( data = rp[causal_gene == T ], aes(geneID,  PROTEIN_ZSCORE), colour = &quot;#FB9A99&quot; , size = 2)+
  geom_point( data = rp[cat3 == T ], aes(geneID,  PROTEIN_ZSCORE ), size = 1, colour = &quot;darkgreen&quot;, shape = 8)+
  theme_bw()+
  theme( legend.title = element_blank(),
         axis.title.x = element_blank(),
         plot.title = element_text(face=&quot;bold&quot;,  size=14, hjust = 0.5),
         axis.title.y = element_text(face=&quot;bold&quot;,  size=10, hjust = 0.5),
         axis.text.x = element_text(face=&quot;bold&quot;,  size=9, angle = 45, hjust = 1),
         axis.text.y = element_text(face=&quot;bold&quot;,  size=9),
         legend.position = &quot;none&quot;)+
  facet_grid(~CATEGORY, scales = &quot;free_x&quot;)</code></pre>
<pre class="r"><code>outrider &lt;- ggplot(rp, aes(geneID,  RNA_ZSCORE ))+
  geom_hline(yintercept = 0, color = &quot;grey50&quot;) +
  geom_quasirandom(data = rp[causal_gene != T ] , aes(geneID,  RNA_ZSCORE),color = &quot;grey80&quot; , size=0.5)+
  stat_summary(fun.data = mean_sdl, fun.args = list(mult=1), geom=&quot;errorbar&quot;, color=&quot;black&quot;, width=0.2) +
  stat_summary(fun = mean, geom=&quot;point&quot;, color=&quot;black&quot;) +
  ylab(&quot;OUTRIDER \n RNA zScore&quot;) + 
  geom_point( data = rp[causal_gene == T ], aes(geneID,  RNA_ZSCORE), color = &quot;#A6CEE3&quot; , size = 2)+
  geom_point( data = rp[cat3 == T ], aes(geneID,  RNA_ZSCORE ), size = 1, colour = &quot;darkgreen&quot;, shape = 8)+
  theme_bw()+
  theme( legend.title = element_blank(),
         axis.title.x = element_blank(),
         plot.title = element_text(face=&quot;bold&quot;,  size=14, hjust = 0.5),
         axis.title.y = element_text(face=&quot;bold&quot;,  size=10, hjust = 0.5),
         axis.text.x = element_text(face=&quot;bold&quot;,  size=9, angle = 45, hjust = 1),
         axis.text.y = element_text(face=&quot;bold&quot;,  size=9),
         legend.position = &quot;none&quot;)+
  facet_grid(~CATEGORY, scales = &quot;free_x&quot;)</code></pre>
<pre class="r"><code>s_fig &lt;- protrider / outrider
s_fig</code></pre>
<p><img src="Scripts_Supplementary_figures_S_Fig1_f_validation_files/figure-html/unnamed-chunk-4-1.png" width="960" /></p>
<pre class="r"><code>pdf(snakemake@output$fig, # &quot;/s/project/mitoMultiOmics/multiOMICs_integration/Figures/Supplementary_figures/S_Fig1_f.pdf&quot;,  
    width = 10, height =7,  useDingbats=FALSE )
print(s_fig) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>

<div id="rmd-source-code">IyctLS0KIycgdGl0bGU6IFN1cHBsZW1lbnRhcnkgRmlnIDFmIHZhbGlkYXRpb24gb2YgcG9zaXRpdmUgY29udHJvbHMgYW5kIFZVUwojJyBhdXRob3I6IHNtaXJub3ZkCiMnIHdiOgojJyAgaW5wdXQ6IAojJyAgLSBjb25maWc6ICdzcmMvY29uZmlnLlInCiMnICAtIHBhdGllbnRfb21pY3M6ICdgc20gY29uZmlnWyJQUk9DX0RBVEEiXSArICIvaW50ZWdyYXRpb24vcGF0aWVudF9vbWljcy5SRFMiYCcKIycgIC0gc2FtcGxlX2Fubm90YXRpb246ICdgc20gY29uZmlnWyJBTk5PVEFUSU9OIl1gJwojJyAgb3V0cHV0OgojJyAgLSBmaWc6ICdgc20gY29uZmlnWyJGSUdVUkVfRElSIl0gKyAiL1N1cHBsZW1lbnRhcnlfZmlndXJlcy9TX0ZpZzFfZi5wZGYiYCcKIycgb3V0cHV0OiAKIycgICBodG1sX2RvY3VtZW50OgojJyAgICBjb2RlX2ZvbGRpbmc6IGhpZGUKIycgICAgY29kZV9kb3dubG9hZDogVFJVRQojJy0tLQoKCiMgbG9hZCBmdW5jdGlvbnMKc291cmNlKHNuYWtlbWFrZUBpbnB1dCRjb25maWcpCgpkZXYub2ZmKCkKCgojIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjCiMgUmVhZCBpbnRlZ3JhdGVkIG9taWNzIGZpbGUgCiMgcnAgPC0gcmVhZFJEUygiL3MvcHJvamVjdC9taXRvTXVsdGlPbWljcy9tdWx0aU9NSUNzX2ludGVncmF0aW9uL3Byb2Nlc3NlZF9kYXRhL2ludGVncmF0aW9uL3BhdGllbnRfb21pY3MuUkRTIikgJT4lIGFzLmRhdGEudGFibGUoKQpycCA8LSByZWFkUkRTKHNuYWtlbWFrZUBpbnB1dCRwYXRpZW50X29taWNzKSAlPiUgYXMuZGF0YS50YWJsZSgpCgoKCiMgTG9hZCBzYW1wbGUgYW5ub3RhdGlvbgojIHNhIDwtIGZyZWFkKCcvcy9wcm9qZWN0L21pdG9NdWx0aU9taWNzL211bHRpT01JQ3NfaW50ZWdyYXRpb24vcmF3X2RhdGEvcHJvdGVvbWljc19hbm5vdGF0aW9uLnRzdicpCnNhIDwtIGZyZWFkKHNuYWtlbWFrZUBpbnB1dCRzYW1wbGVfYW5ub3RhdGlvbikKc2EgPC0gc2FbVVNFX0ZPUl9QUk9URU9NSUNTX1BBUEVSID09IFRdCnNhJHNhbXBsZV9nZW5lIDwtIHBhc3RlMChzYSRTQU1QTEVfSUQsICJfIiwgc2EkS05PV05fTVVUQVRJT04pCnNhIDwtIHNhW29yZGVyKENBVEVHT1JZLCBLTk9XTl9NVVRBVElPTildIAoKCiMgUmVtb3ZlIFVRQ1JGUzEgYXMgc29vbiBhcyBpdCB3YXMgbm90IGRldGVjdGVkIGJ5IHByb3Rlb21pY3MgCiMgYW5kIFJOQS1zZXEgaXMgbm90IGF2YWxpYWJsZSBmb3IgdGhlIGNhc2Ugd2l0aCBrbm93biBwYXRob2dlbmljIHZhaWFudCAKc2EgPC0gc2FbS05PV05fTVVUQVRJT04gIT0gIlVRQ1JGUzEiXQoKCiMgU3Vic2V0IGNhc2VzIHdpdGggcGF0aG9nZW5pYyB2YXJpYW50cyAoSSkgYW5kIFZVUyAoSUkpCnNhIDwtIHNhW0NBVEVHT1JZICVpbiUgYyggIkkiLCAiSUlhIiwgIklJYiIsICJJSWMiKSBdCnNhW0NBVEVHT1JZICVpbiUgYygiSUlhIiwgIklJYiIsICJJSWMiKSwgQ0FURUdPUlkgOj0gIklJIl0Kc2EkQ0FURUdPUlkgPC0gZmFjdG9yKHNhJENBVEVHT1JZLCBjKCAiSSIsICJJSSIpKQoKCiMgU3Vic2V0IGRhdGEgZm9yIGNhdXNhbCBnZW5lcyBpbiBhbGwgc2FtcGxlcwpycCA8LSBycFsgZ2VuZUlEICVpbiUgc2EkS05PV05fTVVUQVRJT05dCgoKCiMgQUNBRDkgYXBwZWFycyBpbiBib3RoIGNhdGVnb3JpZXMsIGR1cGxpY2F0ZSB0aGUgZW50cnkKc2FbS05PV05fTVVUQVRJT04gPT0gIkFDQUQ5IiAmIFNBTVBMRV9JRCA9PSAiT002NTcwOCIsIEtOT1dOX01VVEFUSU9OIDo9ICIgQUNBRDkiXQpzYSRzYW1wbGVfZ2VuZSA8LSBwYXN0ZTAoc2EkU0FNUExFX0lELCAiXyIsIHNhJEtOT1dOX01VVEFUSU9OKQoKcnBfYWNhZDkgPC0gcnBbZ2VuZUlEID09ICJBQ0FEOSIgJiBTQU1QTEVfSUQgIT0gIk9NNjU3MjgiXQpycF9hY2FkOSRnZW5lSUQgPC0gcGFzdGUwKCIgIiwgcnBfYWNhZDkkZ2VuZUlEKQpycCA8LSBycFsgIShnZW5lSUQgPT0gIkFDQUQ5IiAmIFNBTVBMRV9JRCA9PSAiT002NTcwOCIpIF0KcnAgPC0gcmJpbmQocnAsIHJwX2FjYWQ5ICkKcnAkc2FtcGxlX2dlbmUgPC0gcGFzdGUwKHJwJFNBTVBMRV9JRCwgIl8iLCBycCRnZW5lSUQpCnJtKHJwX2FjYWQ5KQoKCiMgQW5ub3RhdGUgd2l0aCBjYXRlZ29yeQpnZW5lX2NhdCA8LSBzYVsgLCBjKCJLTk9XTl9NVVRBVElPTiIsICJDQVRFR09SWSIpXQpnZW5lX2NhdCA8LSBnZW5lX2NhdFshZHVwbGljYXRlZChnZW5lX2NhdCldCnJwIDwtIG1lcmdlKHJwLCBnZW5lX2NhdCwgYnkueCA9ICJnZW5lSUQiLCBieS55ID0gIktOT1dOX01VVEFUSU9OIikKcm0oZ2VuZV9jYXQpCgoKIyBSZW5hbWUgY2F0ZWdvcmllcyAKcnBbQ0FURUdPUlkgPT0gIkkiLCBDQVRFR09SWSA6PSAiUGF0aG9nZW5pYyB2YXJpYW50cyJdCnJwW0NBVEVHT1JZID09ICJJSSIsIENBVEVHT1JZIDo9ICJWVVMiXQoKCiMgQW5ub3RhdGUgd2l0aCBjYXVzYWwgZ2VuZQpycCRjYXVzYWxfZ2VuZSA8LSBOVUxMCnJwWywgY2F1c2FsX2dlbmUgOj0gIHNhbXBsZV9nZW5lICVpbiUgc2Ekc2FtcGxlX2dlbmVdCnJwWywgY2F0MyA6PSBzYW1wbGVfZ2VuZSAlaW4lIHNhW0NBVEVHT1JZID09ICJJSUkiXSRzYW1wbGVfZ2VuZSBdCgpycFtnZW5lSUQgPT0gIlNUSU0xIiwgZ2VuZUlEIDo9ICJTVElNMSoiXQpycFtjYXVzYWxfZ2VuZSA9PSBUICYgaXMubmEoUFJPVEVJTl9aU0NPUkUpLCBQUk9URUlOX1pTQ09SRSA6PSAtSW5mXQpycFtjYXVzYWxfZ2VuZSA9PSBUICYgaXMubmEoUk5BX1pTQ09SRSksIFJOQV9aU0NPUkUgOj0gLUluZl0KCnJwX2MgPC0gcnBbY2F1c2FsX2dlbmUgPT0gVCAmIENBVEVHT1JZICVpbiUgYygiUGF0aG9nZW5pYyB2YXJpYW50cyIgLCAgIlZVUyIpXQpycCRnZW5lSUQgPC0gZmFjdG9yKHJwJGdlbmVJRCwgdW5pcXVlKHJwX2NbIG9yZGVyKFBST1RFSU5fWlNDT1JFKV0kZ2VuZUlEKSkKCgojIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMKCgojKyBmaWcud2lkdGg9MTAsIGZpZy5oZWlnaHQ9NApwcm90cmlkZXIgPC0gZ2dwbG90KHJwLCBhZXMoZ2VuZUlELCAgUFJPVEVJTl9aU0NPUkUgKSkrCiAgZ2VvbV9obGluZSh5aW50ZXJjZXB0ID0gMCwgY29sb3IgPSAiZ3JleTUwIikgKwogIGdlb21fcXVhc2lyYW5kb20oZGF0YSA9IHJwW2NhdXNhbF9nZW5lICE9IFQgXSAsIGFlcyhnZW5lSUQsICBQUk9URUlOX1pTQ09SRSksY29sb3IgPSAiZ3JleTgwIiAsIHNpemU9MC41KSsKICBzdGF0X3N1bW1hcnkoZnVuLmRhdGEgPSBtZWFuX3NkbCwgZnVuLmFyZ3MgPSBsaXN0KG11bHQ9MSksIGdlb209ImVycm9yYmFyIiwgY29sb3I9ImJsYWNrIiwgd2lkdGg9MC4yKSArCiAgc3RhdF9zdW1tYXJ5KGZ1biA9IG1lYW4sIGdlb209InBvaW50IiwgY29sb3I9ImJsYWNrIikgKwogIHlsYWIoIlBST1RSSURFUiBcbiBQcm90ZWluIHpTY29yZSIpICsgCiAgZ2VvbV9wb2ludCggZGF0YSA9IHJwW2NhdXNhbF9nZW5lID09IFQgXSwgYWVzKGdlbmVJRCwgIFBST1RFSU5fWlNDT1JFKSwgY29sb3VyID0gIiNGQjlBOTkiICwgc2l6ZSA9IDIpKwogIGdlb21fcG9pbnQoIGRhdGEgPSBycFtjYXQzID09IFQgXSwgYWVzKGdlbmVJRCwgIFBST1RFSU5fWlNDT1JFICksIHNpemUgPSAxLCBjb2xvdXIgPSAiZGFya2dyZWVuIiwgc2hhcGUgPSA4KSsKICB0aGVtZV9idygpKwogIHRoZW1lKCBsZWdlbmQudGl0bGUgPSBlbGVtZW50X2JsYW5rKCksCiAgICAgICAgIGF4aXMudGl0bGUueCA9IGVsZW1lbnRfYmxhbmsoKSwKICAgICAgICAgcGxvdC50aXRsZSA9IGVsZW1lbnRfdGV4dChmYWNlPSJib2xkIiwgIHNpemU9MTQsIGhqdXN0ID0gMC41KSwKICAgICAgICAgYXhpcy50aXRsZS55ID0gZWxlbWVudF90ZXh0KGZhY2U9ImJvbGQiLCAgc2l6ZT0xMCwgaGp1c3QgPSAwLjUpLAogICAgICAgICBheGlzLnRleHQueCA9IGVsZW1lbnRfdGV4dChmYWNlPSJib2xkIiwgIHNpemU9OSwgYW5nbGUgPSA0NSwgaGp1c3QgPSAxKSwKICAgICAgICAgYXhpcy50ZXh0LnkgPSBlbGVtZW50X3RleHQoZmFjZT0iYm9sZCIsICBzaXplPTkpLAogICAgICAgICBsZWdlbmQucG9zaXRpb24gPSAibm9uZSIpKwogIGZhY2V0X2dyaWQofkNBVEVHT1JZLCBzY2FsZXMgPSAiZnJlZV94IikKCgojKyBmaWcud2lkdGg9MTAsIGZpZy5oZWlnaHQ9NApvdXRyaWRlciA8LSBnZ3Bsb3QocnAsIGFlcyhnZW5lSUQsICBSTkFfWlNDT1JFICkpKwogIGdlb21faGxpbmUoeWludGVyY2VwdCA9IDAsIGNvbG9yID0gImdyZXk1MCIpICsKICBnZW9tX3F1YXNpcmFuZG9tKGRhdGEgPSBycFtjYXVzYWxfZ2VuZSAhPSBUIF0gLCBhZXMoZ2VuZUlELCAgUk5BX1pTQ09SRSksY29sb3IgPSAiZ3JleTgwIiAsIHNpemU9MC41KSsKICBzdGF0X3N1bW1hcnkoZnVuLmRhdGEgPSBtZWFuX3NkbCwgZnVuLmFyZ3MgPSBsaXN0KG11bHQ9MSksIGdlb209ImVycm9yYmFyIiwgY29sb3I9ImJsYWNrIiwgd2lkdGg9MC4yKSArCiAgc3RhdF9zdW1tYXJ5KGZ1biA9IG1lYW4sIGdlb209InBvaW50IiwgY29sb3I9ImJsYWNrIikgKwogIHlsYWIoIk9VVFJJREVSIFxuIFJOQSB6U2NvcmUiKSArIAogIGdlb21fcG9pbnQoIGRhdGEgPSBycFtjYXVzYWxfZ2VuZSA9PSBUIF0sIGFlcyhnZW5lSUQsICBSTkFfWlNDT1JFKSwgY29sb3IgPSAiI0E2Q0VFMyIgLCBzaXplID0gMikrCiAgZ2VvbV9wb2ludCggZGF0YSA9IHJwW2NhdDMgPT0gVCBdLCBhZXMoZ2VuZUlELCAgUk5BX1pTQ09SRSApLCBzaXplID0gMSwgY29sb3VyID0gImRhcmtncmVlbiIsIHNoYXBlID0gOCkrCiAgdGhlbWVfYncoKSsKICB0aGVtZSggbGVnZW5kLnRpdGxlID0gZWxlbWVudF9ibGFuaygpLAogICAgICAgICBheGlzLnRpdGxlLnggPSBlbGVtZW50X2JsYW5rKCksCiAgICAgICAgIHBsb3QudGl0bGUgPSBlbGVtZW50X3RleHQoZmFjZT0iYm9sZCIsICBzaXplPTE0LCBoanVzdCA9IDAuNSksCiAgICAgICAgIGF4aXMudGl0bGUueSA9IGVsZW1lbnRfdGV4dChmYWNlPSJib2xkIiwgIHNpemU9MTAsIGhqdXN0ID0gMC41KSwKICAgICAgICAgYXhpcy50ZXh0LnggPSBlbGVtZW50X3RleHQoZmFjZT0iYm9sZCIsICBzaXplPTksIGFuZ2xlID0gNDUsIGhqdXN0ID0gMSksCiAgICAgICAgIGF4aXMudGV4dC55ID0gZWxlbWVudF90ZXh0KGZhY2U9ImJvbGQiLCAgc2l6ZT05KSwKICAgICAgICAgbGVnZW5kLnBvc2l0aW9uID0gIm5vbmUiKSsKICBmYWNldF9ncmlkKH5DQVRFR09SWSwgc2NhbGVzID0gImZyZWVfeCIpCgoKCiMrIGZpZy53aWR0aD0xMCwgZmlnLmhlaWdodD03CnNfZmlnIDwtIHByb3RyaWRlciAvIG91dHJpZGVyCnNfZmlnCgoKCgpwZGYoc25ha2VtYWtlQG91dHB1dCRmaWcsICMgIi9zL3Byb2plY3QvbWl0b011bHRpT21pY3MvbXVsdGlPTUlDc19pbnRlZ3JhdGlvbi9GaWd1cmVzL1N1cHBsZW1lbnRhcnlfZmlndXJlcy9TX0ZpZzFfZi5wZGYiLCAgCiAgICB3aWR0aCA9IDEwLCBoZWlnaHQgPTcsICB1c2VEaW5nYmF0cz1GQUxTRSApCnByaW50KHNfZmlnKSAKZGV2Lm9mZigpCgoK</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("S_Fig1_f_validation.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
