<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-02-15" />

<title>Enrichment of rare variants in outliers</title>

<script src="libR/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>
<script src="libR/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="libR/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libR/datatables-binding-0.15/datatables.js"></script>
<link href="libR/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libR/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libR/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="libR/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="libR/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Enrichment of rare variants in outliers</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-02-15</h4>

</div>


<pre class="r"><code># Load plotting config and functions
source(&#39;src/config.R&#39;)
source(&quot;src/functions/Integration/integrate_annotate_omics.R&quot;)
source(&quot;src/functions/variant_enrichment.R&quot;)


# Read integrated omics file 
# rp &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics_full.RDS&quot;) %&gt;% as.data.table()
rp &lt;- readRDS(snakemake@input$patient_omics) %&gt;% as.data.table()

# Subset cases with WES and RNA-seq data avaliable 
rp &lt;- rp[WES_avaliable == T &amp; RNA_seq_avaliable == T]


#Keep only genes, with both RNA and protein measured 
rp &lt;- rp[ gene_detected == &quot;RNA and protein detected&quot;]

# Filter for the genes, detected as outliers at least once
outliers &lt;- unique(rp[outlier_class != &quot;non_outlier&quot;]$geneID)
rp &lt;- rp[geneID %in% outliers]

# Add up- down- outlier class
rp &lt;- add_up_down_class(rp, 
                        Padj_threshold =  PADJ_THRESHOLD,
                        Zscore_threshold = ZSCORE_THRESHOLD, 
                        l2FC_threshold = LOG2FC_THRESHOLD)


###################################################
# Subset necessary columns
# Add variant type annotation
rp[ , var_type := &quot;No rare variant&quot;] 
rp[ rare == T , var_type := &quot;Rare variant&quot;] 
rp[ potential_biallelic == T , var_type := &quot;Potential biallelic\n rare variant&quot;] 


# Subset necessary columns
rp &lt;- rp[ , c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;, &quot;var_type&quot;, &quot;outlier_class&quot;, &quot;up_down_outlier&quot;) ]
rp &lt;- rp[!duplicated(rp), ]


###################################################

# As soon as it is not allowed to share genetic data
# Shared data doesn&#39;t allow to reproduce original figure 
# Code below was used to produse figure on the full datset 


# Read variant annotation 

rp &lt;- rp[ , c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;,  &quot;outlier_class&quot;, &quot;up_down_outlier&quot;) ]
rp &lt;- rp[!duplicated(rp), ]

vt &lt;- fread(&quot;../rare_variants_pb.tsv&quot;)
rp &lt;- rp[SAMPLE_ID %in% unique(vt$SAMPLE_ID) ]
vt &lt;- vt[SAMPLE_ID %in% unique(rp$SAMPLE_ID) ]
vt &lt;- vt[ geneID %in% unique(rp$geneID)]

rp &lt;- merge(vt, rp, by = c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;), all.y = T )
rp[is.na(var_type), var_type := &quot;No rare variant&quot;]



###################################################



paste(&quot;Number of samples with WES export and RNA-seq avaliable:&quot;, uniqueN(rp$SAMPLE_ID))</code></pre>
<pre><code>## [1] &quot;Number of samples with WES export and RNA-seq avaliable: 136&quot;</code></pre>
<pre class="r"><code>rp[ , Ncases_per_gene := .N, by = geneID]
paste(&quot;Minimal number of individuals per gene:&quot; , min(rp$Ncases_per_gene))</code></pre>
<pre><code>## [1] &quot;Minimal number of individuals per gene: 63&quot;</code></pre>
<pre class="r"><code># Subset necessary columns
rp &lt;- rp[ , c(&quot;SAMPLE_ID&quot;, &quot;geneID&quot;, &quot;var_type&quot;, &quot;outlier_class&quot;, &quot;up_down_outlier&quot;) ]
rp &lt;- rp[!duplicated(rp), ]




####################</code></pre>
<div id="outlier-class" class="section level1">
<h1>Outlier class</h1>
<pre class="r"><code>os &lt;- rp[, .N, by = outlier_class]</code></pre>
<div id="number-of-outliers-per-class" class="section level3">
<h3>Number of outliers per class</h3>
<pre class="r"><code>DT::datatable(os, caption = &quot;Outlier counts&quot;,   escape = F)</code></pre>
<div id="htmlwidget-30070d99fe8e9c220624" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-30070d99fe8e9c220624">{"x":{"filter":"none","caption":"<caption>Outlier counts<\/caption>","data":[["1","2","3","4"],["non_outlier","RNA-only","protein-only","RNA-and-protein"],[203314,228,1360,169]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>outlier_class<\/th>\n      <th>N<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># All outliers
enr &lt;- reshape2::dcast(data = rp, geneID +  SAMPLE_ID +  var_type ~ outlier_class , 
                       fun.aggregate = cat.agg)

enrichments &lt;- data.frame()
for (outl in unique(rp$outlier_class)){
  tem_df &lt;- enr
  setnames(tem_df, outl, &quot;outliers&quot;)
  tem_df &lt;- tem_df[, c(&quot;geneID&quot;, &quot;SAMPLE_ID&quot;, &quot;outliers&quot;, &quot;var_type&quot; )]
  feat &lt;- unique(tem_df$var_type)
  tem_df &lt;- dcast(data = tem_df, geneID + SAMPLE_ID  + outliers ~ var_type, fun.aggregate = cat.agg) 
  enrichment &lt;- enrich(tem_df, outlier= &quot;outliers&quot; ,features= feat)
  enrichment$ci_left &lt;- enrichment$Estim - 1.96 * enrichment$Std
  enrichment$ci_right &lt;- enrichment$Estim + 1.96 * enrichment$Std
  enrichment$outlier_class &lt;- rep(outl, nrow(enrichment))
  enrichments &lt;- rbind(enrichments, enrichment)
  enrichment &lt;- NULL
  enr$outliers &lt;- NULL
}

enrichments &lt;- as.data.table(enrichments)
enrichments[, significant := 1 ]
enrichments[enrichments$Pval &gt;= 0.05 , significant := 0.9 ]
nodelist &lt;- levels(enrichments$Cat)
enrichments$Cat &lt;- factor(enrichments$Cat, levels= c(&quot;No rare variant&quot;, &quot;Rare variant&quot;, &quot;Potential biallelic\n rare variant&quot;))</code></pre>
</div>
<div id="outlier-class-1" class="section level2">
<h2>Outlier class</h2>
<pre class="r"><code>ggplot(data = enrichments, aes(x = Cat, y = Estim)) +
  geom_pointrange(aes(x = Cat, ymin = ci_left, ymax = ci_right, colour = outlier_class), position = position_dodge(width = 1)) +
  geom_hline(yintercept = 0) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;No rare variant&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;Rare variant&quot;)-0.5) +
  theme_bw() + 
  xlab(&#39;&#39;) + 
  ylab(&#39;Log odds ratio&#39;)  + 
  scale_alpha(guide = &#39;none&#39;)+
  scale_color_manual(values = outlier_colors)+ 
  coord_flip(ylim = c(min(enrichments[enrichments$Pval &lt;0.05, &#39;ci_left&#39; ] , na.rm = T),  max(enrichments[enrichments$Pval &lt;0.05, &quot;ci_right&quot; ], na.rm = T))) </code></pre>
<p><img src="Scripts_MultiOmics_Variant_analysis_enrichment_analysis_files/figure-html/unnamed-chunk-4-1.png" width="576" /></p>
<pre class="r"><code># write_tsv(enrichments, &quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/variant_tables/enrichment_rare_biallelic.tsv&quot;)
write_tsv(enrichments,  snakemake@output$enrichment_rare_pb)


#####################################</code></pre>
</div>
</div>
<div id="up--down--regulation-outliers" class="section level1">
<h1>Up- down- regulation outliers</h1>
<pre class="r"><code>os &lt;- rp[, .N, by = up_down_outlier]</code></pre>
<div id="number-of-outliers-per-class-1" class="section level3">
<h3>Number of outliers per class</h3>
<pre class="r"><code>DT::datatable(os, caption = &quot;Outlier counts&quot;,   escape = F)</code></pre>
<div id="htmlwidget-3eb8fb9c03cdf5b50007" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-3eb8fb9c03cdf5b50007">{"x":{"filter":"none","caption":"<caption>Outlier counts<\/caption>","data":[["1","2","3","4","5","6","7","8"],["non_outlier","RNA_underexpression","Protein_underexpression","RNA_Protein_underexpression","RNA_overexpression","RNA_Protein_overexpression","Protein_overexpression","RNA_over_Protein_under"],[203314,159,863,150,69,18,497,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>up_down_outlier<\/th>\n      <th>N<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>enr &lt;- reshape2::dcast(data = rp, geneID +  SAMPLE_ID +  var_type ~ up_down_outlier , 
                       fun.aggregate = cat.agg)

enrichments &lt;- data.frame()
for (outl in unique(rp$up_down_outlier)){
  tem_df &lt;- enr
  setnames(tem_df, outl, &quot;outliers&quot;)
  tem_df &lt;- tem_df[, c(&quot;geneID&quot;, &quot;SAMPLE_ID&quot;, &quot;outliers&quot;, &quot;var_type&quot; )]
  feat &lt;- unique(tem_df$var_type)
  tem_df &lt;- dcast(data = tem_df, geneID + SAMPLE_ID  + outliers ~ var_type, fun.aggregate = cat.agg) 
  enrichment &lt;- enrich(tem_df, outlier= &quot;outliers&quot; ,features= feat)
  enrichment$ci_left &lt;- enrichment$Estim - 1.96 * enrichment$Std
  enrichment$ci_right &lt;- enrichment$Estim + 1.96 * enrichment$Std
  enrichment$outlier_class &lt;- rep(outl, nrow(enrichment))
  enrichments &lt;- rbind(enrichments, enrichment)
  enr$outliers &lt;- NULL
}

enrichments &lt;- as.data.table(enrichments)
enrichments[, significant := 1 ]
enrichments[enrichments$Pval &gt;= 0.05 , significant := 0.9 ]
nodelist &lt;- levels(enrichments$Cat)
enrichments$Cat &lt;- factor(enrichments$Cat, levels= c(&quot;No rare variant&quot;, &quot;Rare variant&quot;, &quot;Potential biallelic\n rare variant&quot;))</code></pre>
</div>
<div id="outlier-class-2" class="section level2">
<h2>Outlier class</h2>
<pre class="r"><code>ggplot(data = enrichments[outlier_class %in% c(&quot;RNA_underexpression&quot;, &quot;Protein_underexpression&quot;, &quot;RNA_Protein_underexpression&quot;, &quot;non_outlier&quot;)], aes(x = Cat, y = Estim)) +
  geom_pointrange(aes(x = Cat, ymin = ci_left, ymax = ci_right, colour = outlier_class, alpha = significant), position = position_dodge(width = 1)) +
  geom_hline(yintercept = 0) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;No rare variant&quot;)-0.5) +
  geom_vline(color = &quot;gray80&quot;, linetype = &quot;dashed&quot;, xintercept = which(nodelist==&quot;Rare variant&quot;)-0.5) +
  theme_bw() + 
  xlab(&#39;&#39;) + 
  ylab(&#39;Log odds ratio&#39;)  + 
  scale_alpha(guide = &#39;none&#39;)+
  scale_color_manual(values = outlier_colors)+ 
  coord_flip(ylim = c(min(enrichments[enrichments$Pval &lt;0.05, &#39;ci_left&#39; ] , na.rm = T),  max(enrichments[enrichments$Pval &lt;0.05, &quot;ci_right&quot; ], na.rm = T))) </code></pre>
<p><img src="Scripts_MultiOmics_Variant_analysis_enrichment_analysis_files/figure-html/unnamed-chunk-7-1.png" width="576" /></p>
</div>
</div>

<div id="rmd-source-code">#'---
#' title: Enrichment of rare variants in outliers  
#' author: smirnovd
#' wb:
#'  input:
#'  - patient_omics: '`sm config["PROC_DATA"] + "/integration/patient_omics_full.RDS"`'
#'  output:
#'  - enrichment_rare_pb: '`sm config["PROC_DATA"] + "/variant_tables/enrichment_rare_biallelic.tsv"`'
#' output: 
#'   html_document:
#'    code_folding: hide
#'    code_download: TRUE
#'---

# Load plotting config and functions
source('src/config.R')
source("src/functions/Integration/integrate_annotate_omics.R")
source("src/functions/variant_enrichment.R")


# Read integrated omics file 
# rp <- readRDS("/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics_full.RDS") %>% as.data.table()
rp <- readRDS(snakemake@input$patient_omics) %>% as.data.table()

# Subset cases with WES and RNA-seq data avaliable 
rp <- rp[WES_avaliable == T & RNA_seq_avaliable == T]


#Keep only genes, with both RNA and protein measured 
rp <- rp[ gene_detected == "RNA and protein detected"]

# Filter for the genes, detected as outliers at least once
outliers <- unique(rp[outlier_class != "non_outlier"]$geneID)
rp <- rp[geneID %in% outliers]

# Add up- down- outlier class
rp <- add_up_down_class(rp, 
                        Padj_threshold =  PADJ_THRESHOLD,
                        Zscore_threshold = ZSCORE_THRESHOLD, 
                        l2FC_threshold = LOG2FC_THRESHOLD)


###################################################
# Subset necessary columns
# Add variant type annotation
rp[ , var_type := "No rare variant"] 
rp[ rare == T , var_type := "Rare variant"] 
rp[ potential_biallelic == T , var_type := "Potential biallelic\n rare variant"] 


# Subset necessary columns
rp <- rp[ , c("SAMPLE_ID", "geneID", "var_type", "outlier_class", "up_down_outlier") ]
rp <- rp[!duplicated(rp), ]


###################################################

# As soon as it is not allowed to share genetic data
# Shared data doesn't allow to reproduce original figure 
# Code below was used to produse figure on the full datset 


# Read variant annotation 

rp <- rp[ , c("SAMPLE_ID", "geneID",  "outlier_class", "up_down_outlier") ]
rp <- rp[!duplicated(rp), ]

vt <- fread("../rare_variants_pb.tsv")
rp <- rp[SAMPLE_ID %in% unique(vt$SAMPLE_ID) ]
vt <- vt[SAMPLE_ID %in% unique(rp$SAMPLE_ID) ]
vt <- vt[ geneID %in% unique(rp$geneID)]

rp <- merge(vt, rp, by = c("SAMPLE_ID", "geneID"), all.y = T )
rp[is.na(var_type), var_type := "No rare variant"]



###################################################



paste("Number of samples with WES export and RNA-seq avaliable:", uniqueN(rp$SAMPLE_ID))
rp[ , Ncases_per_gene := .N, by = geneID]
paste("Minimal number of individuals per gene:" , min(rp$Ncases_per_gene))
# Subset necessary columns
rp <- rp[ , c("SAMPLE_ID", "geneID", "var_type", "outlier_class", "up_down_outlier") ]
rp <- rp[!duplicated(rp), ]




####################

#' # Outlier class
os <- rp[, .N, by = outlier_class]

#' ### Number of outliers per class
DT::datatable(os, caption = "Outlier counts",   escape = F)


# All outliers
enr <- reshape2::dcast(data = rp, geneID +  SAMPLE_ID +  var_type ~ outlier_class , 
                       fun.aggregate = cat.agg)

enrichments <- data.frame()
for (outl in unique(rp$outlier_class)){
  tem_df <- enr
  setnames(tem_df, outl, "outliers")
  tem_df <- tem_df[, c("geneID", "SAMPLE_ID", "outliers", "var_type" )]
  feat <- unique(tem_df$var_type)
  tem_df <- dcast(data = tem_df, geneID + SAMPLE_ID  + outliers ~ var_type, fun.aggregate = cat.agg) 
  enrichment <- enrich(tem_df, outlier= "outliers" ,features= feat)
  enrichment$ci_left <- enrichment$Estim - 1.96 * enrichment$Std
  enrichment$ci_right <- enrichment$Estim + 1.96 * enrichment$Std
  enrichment$outlier_class <- rep(outl, nrow(enrichment))
  enrichments <- rbind(enrichments, enrichment)
  enrichment <- NULL
  enr$outliers <- NULL
}

enrichments <- as.data.table(enrichments)
enrichments[, significant := 1 ]
enrichments[enrichments$Pval >= 0.05 , significant := 0.9 ]
nodelist <- levels(enrichments$Cat)
enrichments$Cat <- factor(enrichments$Cat, levels= c("No rare variant", "Rare variant", "Potential biallelic\n rare variant"))


#' ## Outlier class
#+ fig.width=6, fig.height=9
ggplot(data = enrichments, aes(x = Cat, y = Estim)) +
  geom_pointrange(aes(x = Cat, ymin = ci_left, ymax = ci_right, colour = outlier_class), position = position_dodge(width = 1)) +
  geom_hline(yintercept = 0) +
  geom_vline(color = "gray80", linetype = "dashed", xintercept = which(nodelist=="No rare variant")-0.5) +
  geom_vline(color = "gray80", linetype = "dashed", xintercept = which(nodelist=="Rare variant")-0.5) +
  theme_bw() + 
  xlab('') + 
  ylab('Log odds ratio')  + 
  scale_alpha(guide = 'none')+
  scale_color_manual(values = outlier_colors)+ 
  coord_flip(ylim = c(min(enrichments[enrichments$Pval <0.05, 'ci_left' ] , na.rm = T),  max(enrichments[enrichments$Pval <0.05, "ci_right" ], na.rm = T))) 


# write_tsv(enrichments, "/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/variant_tables/enrichment_rare_biallelic.tsv")
write_tsv(enrichments,  snakemake@output$enrichment_rare_pb)


#####################################
#' # Up- down- regulation outliers
os <- rp[, .N, by = up_down_outlier]


#' ### Number of outliers per class
DT::datatable(os, caption = "Outlier counts",   escape = F)

enr <- reshape2::dcast(data = rp, geneID +  SAMPLE_ID +  var_type ~ up_down_outlier , 
                       fun.aggregate = cat.agg)

enrichments <- data.frame()
for (outl in unique(rp$up_down_outlier)){
  tem_df <- enr
  setnames(tem_df, outl, "outliers")
  tem_df <- tem_df[, c("geneID", "SAMPLE_ID", "outliers", "var_type" )]
  feat <- unique(tem_df$var_type)
  tem_df <- dcast(data = tem_df, geneID + SAMPLE_ID  + outliers ~ var_type, fun.aggregate = cat.agg) 
  enrichment <- enrich(tem_df, outlier= "outliers" ,features= feat)
  enrichment$ci_left <- enrichment$Estim - 1.96 * enrichment$Std
  enrichment$ci_right <- enrichment$Estim + 1.96 * enrichment$Std
  enrichment$outlier_class <- rep(outl, nrow(enrichment))
  enrichments <- rbind(enrichments, enrichment)
  enr$outliers <- NULL
}

enrichments <- as.data.table(enrichments)
enrichments[, significant := 1 ]
enrichments[enrichments$Pval >= 0.05 , significant := 0.9 ]
nodelist <- levels(enrichments$Cat)
enrichments$Cat <- factor(enrichments$Cat, levels= c("No rare variant", "Rare variant", "Potential biallelic\n rare variant"))


#' ## Outlier class
#+ fig.width=6, fig.height=9
ggplot(data = enrichments[outlier_class %in% c("RNA_underexpression", "Protein_underexpression", "RNA_Protein_underexpression", "non_outlier")], aes(x = Cat, y = Estim)) +
  geom_pointrange(aes(x = Cat, ymin = ci_left, ymax = ci_right, colour = outlier_class, alpha = significant), position = position_dodge(width = 1)) +
  geom_hline(yintercept = 0) +
  geom_vline(color = "gray80", linetype = "dashed", xintercept = which(nodelist=="No rare variant")-0.5) +
  geom_vline(color = "gray80", linetype = "dashed", xintercept = which(nodelist=="Rare variant")-0.5) +
  theme_bw() + 
  xlab('') + 
  ylab('Log odds ratio')  + 
  scale_alpha(guide = 'none')+
  scale_color_manual(values = outlier_colors)+ 
  coord_flip(ylim = c(min(enrichments[enrichments$Pval <0.05, 'ci_left' ] , na.rm = T),  max(enrichments[enrichments$Pval <0.05, "ci_right" ], na.rm = T))) 

</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("enrichment_analysis.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
