<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="smirnovd" />

<meta name="date" content="2021-01-10" />

<title>Figure 3 A C E</title>

<script src="libR/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="libR/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="libR/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="libR/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="libR/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="libR/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="libR/tocify-1.9.1/jquery.tocify.js"></script>
<script src="libR/navigation-1.1/tabsets.js"></script>
<script src="libR/navigation-1.1/codefolding.js"></script>
<script src="libR/navigation-1.1/sourceembed.js"></script>
<link href="libR/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="libR/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="lib/add_content_table.css" type="text/css" />
<link rel="stylesheet" href="lib/leo_style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
<li role="separator" class="divider"></li>
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Figure 3 A C E</h1>
<h4 class="author">smirnovd</h4>
<h4 class="date">2021-01-10</h4>

</div>


<pre class="r"><code>dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>source(snakemake@input$config)


# Add necessaries for gene structures
suppressPackageStartupMessages({
  stringsAsFactors=F
  library(Gviz)
  library(GenomicRanges)
  library(ggbio)
  library(EnsDb.Hsapiens.v75)
})

## Making a short cut to ensembl 
edb &lt;- EnsDb.Hsapiens.v75


# Load sample annotation
# sa &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/filemappings/proteomics_annotation.tsv&#39;)
sa &lt;- fread(snakemake@input$sample_annotation)
sa &lt;- sa[USE_FOR_PROTEOMICS_PAPER == T]

# Read integrated omics file 
# rp &lt;- readRDS(&quot;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics_full.RDS&quot;) %&gt;% as.data.table()
rp &lt;- readRDS(snakemake@input$patient_omics) %&gt;% as.data.table()


# Load patient&#39;s affected organs
patients &lt;- fread(snakemake@input$patient_organs)
# patients &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/HPO/Patients_affected_organs.tsv&#39;)



######################################################</code></pre>
<div id="fig3a-mrpl38" class="section level3">
<h3>Fig3a MRPL38</h3>
<pre class="r"><code>samp &lt;- rp[ SAMPLE_ID == &quot;OM57837&quot; &amp; outlier_class != &quot;non_outlier&quot;]

fig3a &lt;- ggplot(samp[gene_detected == &quot;RNA and protein detected&quot; ], aes(RNA_ZSCORE, PROTEIN_ZSCORE)) +
  geom_vline(xintercept = 0, color = &quot;grey50&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey50&quot;) +
  
  geom_point(aes(color= outlier_class, shape= gene_class, size = Semantic_sim )) + 
  
  xlab(&quot;RNA zScore&quot;) + 
  ylab(&quot;Protein zScore&quot;) +
  scale_color_manual(breaks = c(&quot;non_outlier&quot;,&quot;RNA_outlier&quot;,&quot;Protein_outlier&quot;, &quot;RNA_Protein_outlier&quot;),
                     values = outlier_colors) +
  
  scale_shape_manual(values = gene_shapes) +
  
  scale_size(range = c(2.5, 6)) +
  
  geom_text_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot;  &amp; gene_class == &quot;no rare variant&quot; ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= &quot;grey20&quot;, 
                  size= 2.5, show.legend = F)+ 
  
  geom_text_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot;  &amp; gene_class == &quot;1 rare variant&quot; &amp; ( causal_gene ==F | is.na(causal_gene)) ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= &quot;black&quot;, 
                  size= 3.1, show.legend = F) + 
  
  geom_label_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot; &amp; gene_class == &quot;rare pot. biallelic variants&quot; | causal_gene == T], 
                   mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID ), 
                   box.padding = unit(0.3, &quot;lines&quot;), colour= &quot;black&quot;, 
                   size= 4, show.legend = F ) + 
  
  coord_fixed( xlim = c( -7,  6 ),  ylim = c( -7,  6 )) +  
  theme_bw() +
  
  theme( axis.title.x= element_text( size= 10, margin = NULL,face=&quot;bold&quot;),
         axis.title.y= element_text( size= 10, margin = NULL,face=&quot;bold&quot;),
         legend.title = element_blank(),
         axis.text.x = element_text(face=&quot;bold&quot;,  size=10),
         axis.text.y = element_text(face=&quot;bold&quot;,  size=10),
         legend.position = &quot;none&quot;,
         plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;) ) 
fig3a</code></pre>
<p><img src="Scripts_Paper_figures_Figure3_ace_files/figure-html/unnamed-chunk-2-1.png" width="480" /></p>
<pre class="r"><code>pdf(&quot;/s/project/mitoMultiOmics/proteome_analysis/Figures/paper_figures/FIG3a.pdf&quot;,  
    width = 5, height =5,  useDingbats=FALSE )
print(fig3a) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>Gene structure and variant</p>
<pre class="r"><code>autoplot(edb, ~ gene_name == &quot;MRPL38&quot;, stat = &quot;reduce&quot; ,  names.expr=&quot;MRPL38&quot;) + 
  theme_void() + scale_x_reverse() </code></pre>
<p><img src="Scripts_Paper_figures_Figure3_ace_files/figure-html/unnamed-chunk-3-1.png" width="480" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>###############################################################################


# Read CORUM complexes
# corum &lt;- fread(&#39;/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/Complexes/CORUM.tsv&#39;)
corum &lt;- fread(snakemake@input$corum)

## Get Corum MRPL/S complexes
corum &lt;- corum[ COMPLEX %in% c(&quot;28S ribosomal subunit, mitochondrial&quot;, &quot;39S ribosomal subunit, mitochondrial&quot;)]





case_MRPL38 &lt;- rp[SAMPLE_ID == &quot;OM57837&quot;] 
case_MRPL38 &lt;- case_MRPL38[!is.na(PROTEIN_LOG2FC )]

case_MRPL38[, small_sub := geneID %in% corum[COMPLEX == &quot;28S ribosomal subunit, mitochondrial&quot;]$geneID]
case_MRPL38[, large_sub := geneID %in% corum[COMPLEX == &quot;39S ribosomal subunit, mitochondrial&quot;]$geneID]


# Small subunit hilighted in volcano plot
ss &lt;- ggplot(case_MRPL38, aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE) ))+
  geom_vline(xintercept = 0, color = &quot;grey30&quot;) +
  geom_point( color = &quot;gray80&quot;) +
  geom_point( data = case_MRPL38[small_sub == T ],
              aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE)), color = &quot;#edae49&quot; ) + 
  xlab(&quot;log2 fold change&quot;) + 
  ylab(&quot;-log10(p-value)&quot;) +
  ggtitle(&quot;small ribosomal proteins&quot;)+
  scale_x_continuous(limits = c( -max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T), 
                                 max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T) ))+  
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, size=12, color = &quot;#edae49&quot;, face=&quot;bold&quot;),
        axis.title.x= element_text( size=10, margin = NULL,face=&quot;bold&quot;),
        axis.title.y= element_text( size=10, margin = NULL,face=&quot;bold&quot;),
        axis.text.x = element_text(face=&quot;bold&quot;,  size=10),
        axis.text.y = element_text(face=&quot;bold&quot;,  size=10),
        legend.title = element_blank() )

# Large subunit hilighted in volcano plot
ls &lt;- ggplot(case_MRPL38, aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE) ))+
  geom_vline(xintercept = 0, color = &quot;grey30&quot;) +
  geom_point(color = &quot;gray80&quot;) +
  geom_point( data = case_MRPL38[large_sub == T ],
              aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE)), color = &quot;#00798c&quot; ) + 
  geom_point(data = case_MRPL38[ geneID == &quot;MRPL38&quot;], 
             aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE) ), color = &quot;#FB9A99&quot; ) +
  geom_text_repel(data = case_MRPL38[ geneID == &quot;MRPL38&quot;], aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE), label =  geneID), size = 3,fontface =&quot;bold&quot; ) +
  xlab(&quot;log2 fold change&quot;) + 
  ylab(&quot;-log10(p-value)&quot;) +
  ggtitle(&quot;large ribosomal proteins&quot;)+
  
  scale_x_continuous(limits = c( -max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T), 
                                 max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T) ))+ 
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, size=12, colour = &quot;#00798c&quot;, face=&quot;bold&quot;),
        axis.title.x= element_text( size=10, margin = NULL,face=&quot;bold&quot;),
        axis.title.y= element_text( size=10, margin = NULL,face=&quot;bold&quot;),
        axis.text.x = element_text(face=&quot;bold&quot;,  size=10),
        axis.text.y = element_text(face=&quot;bold&quot;,  size=10),
        legend.title = element_blank() )</code></pre>
<pre class="r"><code>Fig_3c &lt;- ss / ls
Fig_3c</code></pre>
<p><img src="Scripts_Paper_figures_Figure3_ace_files/figure-html/unnamed-chunk-4-1.png" width="480" /></p>
<pre class="r"><code>pdf(&quot;/s/project/mitoMultiOmics/proteome_analysis/Figures/Fig_3c.pdf&quot;,  
    width = 5, height =7,  useDingbats=FALSE )
print(Fig_3c) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<pre class="r"><code>###############################################################################</code></pre>
</div>
<div id="fig3e-lig3" class="section level3">
<h3>Fig3e LIG3</h3>
<pre class="r"><code>samp &lt;- rp[ SAMPLE_ID == &quot;OM91786&quot;  &amp; outlier_class != &quot;non_outlier&quot; ] #subset for patient

fig3e &lt;- ggplot(samp[gene_detected == &quot;RNA and protein detected&quot; ], aes(RNA_ZSCORE, PROTEIN_ZSCORE)) +
  geom_vline(xintercept = 0, color = &quot;grey50&quot;) +
  geom_hline(yintercept = 0, color = &quot;grey50&quot;) +
  
  geom_point(aes(color= outlier_class, shape= gene_class, size = Semantic_sim )) + 
  
  xlab(&quot;RNA zScore&quot;) + 
  ylab(&quot;Protein zScore&quot;) +
  scale_color_manual(breaks = c(&quot;non_outlier&quot;,&quot;RNA_outlier&quot;,&quot;Protein_outlier&quot;, &quot;RNA_Protein_outlier&quot;),
                     values = outlier_colors) +
  
  scale_shape_manual(values = gene_shapes) +
  
  scale_size(range = c(2.5, 6)) +
  
  geom_text_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot;  &amp; gene_class == &quot;no rare variant&quot; ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= &quot;grey20&quot;, 
                  size= 2.5, show.legend = F)+ 
  
  geom_text_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot;  &amp; gene_class == &quot;1 rare variant&quot; &amp; ( causal_gene ==F | is.na(causal_gene)) ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= &quot;black&quot;, 
                  size= 3.1, show.legend = F) + 
  
  geom_label_repel(data= samp[gene_detected == &quot;RNA and protein detected&quot; &amp; gene_class == &quot;rare pot. biallelic variants&quot; | causal_gene == T], 
                   mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID ), 
                   box.padding = unit(0.3, &quot;lines&quot;), colour= &quot;black&quot;, 
                   size= 4, show.legend = F ) + 
  
  coord_fixed( xlim = c( -7, 7 ),  ylim = c( -7,  7 )) +  
  theme_bw() +
  
  theme( axis.title.x= element_text( size= 10, margin = NULL,face=&quot;bold&quot;),
         axis.title.y= element_text( size= 10, margin = NULL,face=&quot;bold&quot;),
         legend.title = element_blank(),
         axis.text.x = element_text(face=&quot;bold&quot;,  size=10),
         axis.text.y = element_text(face=&quot;bold&quot;,  size=10),
         legend.position = &quot;none&quot;,
         plot.margin = margin(0, 0, 0, 0, &quot;cm&quot;) ) 

fig3e</code></pre>
<p><img src="Scripts_Paper_figures_Figure3_ace_files/figure-html/unnamed-chunk-5-1.png" width="480" /></p>
<pre class="r"><code>pdf(&quot;/s/project/mitoMultiOmics/proteome_analysis/Figures/paper_figures/FIG3e.pdf&quot;,  
    width = 5, height =5,  useDingbats=FALSE )
print(fig3e) 
dev.off()</code></pre>
<pre><code>## png 
##   2</code></pre>
<p>Gene structure and variant</p>
<pre class="r"><code>autoplot(edb, ~ gene_name == &quot;LIG3&quot;, stat = &quot;reduce&quot;,  names.expr=&quot;LIG3&quot;) + 
  theme_void()  </code></pre>
<p><img src="Scripts_Paper_figures_Figure3_ace_files/figure-html/unnamed-chunk-6-1.png" width="480" /></p>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>## null device 
##           1</code></pre>
<pre class="r"><code>unloadNamespace(&quot;ggbio&quot;)</code></pre>
</div>

<div id="rmd-source-code">#'---
#' title: Figure 3 A C E 
#' author: smirnovd
#' wb:
#'  input:
#'  - config: 'src/config.R'
#'  - patient_omics: '`sm config["PROC_DATA"] + "/integration/patient_omics_full.RDS"`'
#'  - sample_annotation: '`sm config["ANNOTATION"]`'
#'  - patient_organs: '`sm config["PROC_DATA"] + "/HPO/Patients_affected_organs.tsv"`'
#'  - corum: '`sm config["PROC_DATA"] + "/Complexes/CORUM.tsv"`'
#'  output:
#' output: 
#'   html_document:
#'    code_folding: hide
#'    code_download: TRUE
#'---


dev.off()
source(snakemake@input$config)


# Add necessaries for gene structures
suppressPackageStartupMessages({
  stringsAsFactors=F
  library(Gviz)
  library(GenomicRanges)
  library(ggbio)
  library(EnsDb.Hsapiens.v75)
})

## Making a short cut to ensembl 
edb <- EnsDb.Hsapiens.v75


# Load sample annotation
# sa <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/filemappings/proteomics_annotation.tsv')
sa <- fread(snakemake@input$sample_annotation)
sa <- sa[USE_FOR_PROTEOMICS_PAPER == T]

# Read integrated omics file 
# rp <- readRDS("/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/integration/patient_omics_full.RDS") %>% as.data.table()
rp <- readRDS(snakemake@input$patient_omics) %>% as.data.table()


# Load patient's affected organs
patients <- fread(snakemake@input$patient_organs)
# patients <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/HPO/Patients_affected_organs.tsv')



######################################################
#' ### Fig3a MRPL38
#+ fig.width=5, fig.height=5

samp <- rp[ SAMPLE_ID == "OM57837" & outlier_class != "non_outlier"]

fig3a <- ggplot(samp[gene_detected == "RNA and protein detected" ], aes(RNA_ZSCORE, PROTEIN_ZSCORE)) +
  geom_vline(xintercept = 0, color = "grey50") +
  geom_hline(yintercept = 0, color = "grey50") +
  
  geom_point(aes(color= outlier_class, shape= gene_class, size = Semantic_sim )) + 
  
  xlab("RNA zScore") + 
  ylab("Protein zScore") +
  scale_color_manual(breaks = c("non_outlier","RNA_outlier","Protein_outlier", "RNA_Protein_outlier"),
                     values = outlier_colors) +
  
  scale_shape_manual(values = gene_shapes) +
  
  scale_size(range = c(2.5, 6)) +
  
  geom_text_repel(data= samp[gene_detected == "RNA and protein detected"  & gene_class == "no rare variant" ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= "grey20", 
                  size= 2.5, show.legend = F)+ 
  
  geom_text_repel(data= samp[gene_detected == "RNA and protein detected"  & gene_class == "1 rare variant" & ( causal_gene ==F | is.na(causal_gene)) ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= "black", 
                  size= 3.1, show.legend = F) + 
  
  geom_label_repel(data= samp[gene_detected == "RNA and protein detected" & gene_class == "rare pot. biallelic variants" | causal_gene == T], 
                   mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID ), 
                   box.padding = unit(0.3, "lines"), colour= "black", 
                   size= 4, show.legend = F ) + 
  
  coord_fixed( xlim = c( -7,  6 ),  ylim = c( -7,  6 )) +  
  theme_bw() +
  
  theme( axis.title.x= element_text( size= 10, margin = NULL,face="bold"),
         axis.title.y= element_text( size= 10, margin = NULL,face="bold"),
         legend.title = element_blank(),
         axis.text.x = element_text(face="bold",  size=10),
         axis.text.y = element_text(face="bold",  size=10),
         legend.position = "none",
         plot.margin = margin(0, 0, 0, 0, "cm") ) 
fig3a

pdf("/s/project/mitoMultiOmics/proteome_analysis/Figures/paper_figures/FIG3a.pdf",  
    width = 5, height =5,  useDingbats=FALSE )
print(fig3a) 
dev.off()


#' Gene structure and variant 
#+ fig.width=5, fig.height=2
autoplot(edb, ~ gene_name == "MRPL38", stat = "reduce" ,  names.expr="MRPL38") + 
  theme_void() + scale_x_reverse() 
dev.off()


###############################################################################


# Read CORUM complexes
# corum <- fread('/s/project/mitoMultiOmics/multiOMICs_integration/processed_data/Complexes/CORUM.tsv')
corum <- fread(snakemake@input$corum)

## Get Corum MRPL/S complexes
corum <- corum[ COMPLEX %in% c("28S ribosomal subunit, mitochondrial", "39S ribosomal subunit, mitochondrial")]





case_MRPL38 <- rp[SAMPLE_ID == "OM57837"] 
case_MRPL38 <- case_MRPL38[!is.na(PROTEIN_LOG2FC )]

case_MRPL38[, small_sub := geneID %in% corum[COMPLEX == "28S ribosomal subunit, mitochondrial"]$geneID]
case_MRPL38[, large_sub := geneID %in% corum[COMPLEX == "39S ribosomal subunit, mitochondrial"]$geneID]


# Small subunit hilighted in volcano plot
ss <- ggplot(case_MRPL38, aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE) ))+
  geom_vline(xintercept = 0, color = "grey30") +
  geom_point( color = "gray80") +
  geom_point( data = case_MRPL38[small_sub == T ],
              aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE)), color = "#edae49" ) + 
  xlab("log2 fold change") + 
  ylab("-log10(p-value)") +
  ggtitle("small ribosomal proteins")+
  scale_x_continuous(limits = c( -max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T), 
                                 max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T) ))+  
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, size=12, color = "#edae49", face="bold"),
        axis.title.x= element_text( size=10, margin = NULL,face="bold"),
        axis.title.y= element_text( size=10, margin = NULL,face="bold"),
        axis.text.x = element_text(face="bold",  size=10),
        axis.text.y = element_text(face="bold",  size=10),
        legend.title = element_blank() )

# Large subunit hilighted in volcano plot
ls <- ggplot(case_MRPL38, aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE) ))+
  geom_vline(xintercept = 0, color = "grey30") +
  geom_point(color = "gray80") +
  geom_point( data = case_MRPL38[large_sub == T ],
              aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE)), color = "#00798c" ) + 
  geom_point(data = case_MRPL38[ geneID == "MRPL38"], 
             aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE) ), color = "#FB9A99" ) +
  geom_text_repel(data = case_MRPL38[ geneID == "MRPL38"], aes(PROTEIN_LOG2FC, -log10(PROTEIN_PVALUE), label =  geneID), size = 3,fontface ="bold" ) +
  xlab("log2 fold change") + 
  ylab("-log10(p-value)") +
  ggtitle("large ribosomal proteins")+
  
  scale_x_continuous(limits = c( -max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T), 
                                 max( abs(case_MRPL38$PROTEIN_LOG2FC) , na.rm = T) ))+ 
  theme_bw()+
  theme(plot.title = element_text(hjust = 0.5, size=12, colour = "#00798c", face="bold"),
        axis.title.x= element_text( size=10, margin = NULL,face="bold"),
        axis.title.y= element_text( size=10, margin = NULL,face="bold"),
        axis.text.x = element_text(face="bold",  size=10),
        axis.text.y = element_text(face="bold",  size=10),
        legend.title = element_blank() )

#+ fig.width=5, fig.height=7
Fig_3c <- ss / ls
Fig_3c


pdf("/s/project/mitoMultiOmics/proteome_analysis/Figures/Fig_3c.pdf",  
    width = 5, height =7,  useDingbats=FALSE )
print(Fig_3c) 
dev.off()





###############################################################################

#' ### Fig3e LIG3
#+ fig.width=5, fig.height=5
samp <- rp[ SAMPLE_ID == "OM91786"  & outlier_class != "non_outlier" ] #subset for patient

fig3e <- ggplot(samp[gene_detected == "RNA and protein detected" ], aes(RNA_ZSCORE, PROTEIN_ZSCORE)) +
  geom_vline(xintercept = 0, color = "grey50") +
  geom_hline(yintercept = 0, color = "grey50") +
  
  geom_point(aes(color= outlier_class, shape= gene_class, size = Semantic_sim )) + 
  
  xlab("RNA zScore") + 
  ylab("Protein zScore") +
  scale_color_manual(breaks = c("non_outlier","RNA_outlier","Protein_outlier", "RNA_Protein_outlier"),
                     values = outlier_colors) +
  
  scale_shape_manual(values = gene_shapes) +
  
  scale_size(range = c(2.5, 6)) +
  
  geom_text_repel(data= samp[gene_detected == "RNA and protein detected"  & gene_class == "no rare variant" ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= "grey20", 
                  size= 2.5, show.legend = F)+ 
  
  geom_text_repel(data= samp[gene_detected == "RNA and protein detected"  & gene_class == "1 rare variant" & ( causal_gene ==F | is.na(causal_gene)) ], 
                  mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID), colour= "black", 
                  size= 3.1, show.legend = F) + 
  
  geom_label_repel(data= samp[gene_detected == "RNA and protein detected" & gene_class == "rare pot. biallelic variants" | causal_gene == T], 
                   mapping=aes(RNA_ZSCORE, PROTEIN_ZSCORE, label= geneID ), 
                   box.padding = unit(0.3, "lines"), colour= "black", 
                   size= 4, show.legend = F ) + 
  
  coord_fixed( xlim = c( -7, 7 ),  ylim = c( -7,  7 )) +  
  theme_bw() +
  
  theme( axis.title.x= element_text( size= 10, margin = NULL,face="bold"),
         axis.title.y= element_text( size= 10, margin = NULL,face="bold"),
         legend.title = element_blank(),
         axis.text.x = element_text(face="bold",  size=10),
         axis.text.y = element_text(face="bold",  size=10),
         legend.position = "none",
         plot.margin = margin(0, 0, 0, 0, "cm") ) 

fig3e

pdf("/s/project/mitoMultiOmics/proteome_analysis/Figures/paper_figures/FIG3e.pdf",  
    width = 5, height =5,  useDingbats=FALSE )
print(fig3e) 
dev.off()



#' Gene structure and variant 
#+ fig.width=5, fig.height=2
autoplot(edb, ~ gene_name == "LIG3", stat = "reduce",  names.expr="LIG3") + 
  theme_void()  
dev.off()


unloadNamespace("ggbio")
</div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("Figure3_ace.R");
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
